{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# BEDBASE workflow tutorial"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following demo has the purpose of demonstrating how to process, generate statistics and plots of BED files generated by the R package Genomic Distributions using the `bedhost` REST API for the bedstat and bedbuncher pipelines output. \n",
    "\n",
    "Notes:\n",
    "\n",
    "- If this hasn't been already done, we recommend starting this jupyter notebook enabling sudo permissions since steps such as downloading `docker` or running an elasticsearch `docker` container won't be executed otherwise. This can be done with `sudo jupyter notebook --allow-root`\n",
    "\n",
    " \n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "toc": true
   },
   "source": [
    "<h1>Table of Contents<span class=\"tocSkip\"></span></h1>\n",
    "<div class=\"toc\"><ul class=\"toc-item\"><li><span><a href=\"#Create-a-tutorial-directory-and-download-demo-files\" data-toc-modified-id=\"Create-a-tutorial-directory-and-download-demo-files-1\"><span class=\"toc-item-num\">1&nbsp;&nbsp;</span>Create a tutorial directory and download demo files</a></span></li><li><span><a href=\"#Generate-statistics-and-plots-of-BED-files-using-BEDSTAT\" data-toc-modified-id=\"Generate-statistics-and-plots-of-BED-files-using-BEDSTAT-2\"><span class=\"toc-item-num\">2&nbsp;&nbsp;</span>Generate statistics and plots of BED files using BEDSTAT</a></span><ul class=\"toc-item\"><li><span><a href=\"#Create-a-PEP-describing-the-BED-files-to-process\" data-toc-modified-id=\"Create-a-PEP-describing-the-BED-files-to-process-2.1\"><span class=\"toc-item-num\">2.1&nbsp;&nbsp;</span>Create a PEP describing the BED files to process</a></span></li><li><span><a href=\"#Download-bedstat-and-the-Bedbase-configuration-manager-(bbconf)\" data-toc-modified-id=\"Download-bedstat-and-the-Bedbase-configuration-manager-(bbconf)-2.2\"><span class=\"toc-item-num\">2.2&nbsp;&nbsp;</span>Download bedstat and the Bedbase configuration manager (bbconf)</a></span></li><li><span><a href=\"#Inititiate-a-local-elasticsearch-cluster\" data-toc-modified-id=\"Inititiate-a-local-elasticsearch-cluster-2.3\"><span class=\"toc-item-num\">2.3&nbsp;&nbsp;</span>Inititiate a local elasticsearch cluster</a></span></li><li><span><a href=\"#Run-bedstat--on-the-demo-PEP\" data-toc-modified-id=\"Run-bedstat--on-the-demo-PEP-2.4\"><span class=\"toc-item-num\">2.4&nbsp;&nbsp;</span>Run bedstat  on the demo PEP</a></span></li></ul></li><li><span><a href=\"#Create-bedsets-using-BEDBUNCHER\" data-toc-modified-id=\"Create-bedsets-using-BEDBUNCHER-3\"><span class=\"toc-item-num\">3&nbsp;&nbsp;</span>Create bedsets using BEDBUNCHER</a></span><ul class=\"toc-item\"><li><span><a href=\"#Create-a-new-PEP-describing-the-bedset-name-and-specific-JSON-query\" data-toc-modified-id=\"Create-a-new-PEP-describing-the-bedset-name-and-specific-JSON-query-3.1\"><span class=\"toc-item-num\">3.1&nbsp;&nbsp;</span>Create a new PEP describing the bedset name and specific JSON query</a></span></li><li><span><a href=\"#Download-bedbuncher--and-install-CML-dependencies\" data-toc-modified-id=\"Download-bedbuncher--and-install-CML-dependencies-3.2\"><span class=\"toc-item-num\">3.2&nbsp;&nbsp;</span>Download bedbuncher  and install CML dependencies</a></span></li><li><span><a href=\"#Run-bedbuncher-using-Looper\" data-toc-modified-id=\"Run-bedbuncher-using-Looper-3.3\"><span class=\"toc-item-num\">3.3&nbsp;&nbsp;</span>Run bedbuncher using Looper</a></span></li></ul></li><li><span><a href=\"#Run-local-instance-of-the-bedhost-API\" data-toc-modified-id=\"Run-local-instance-of-the-bedhost-API-4\"><span class=\"toc-item-num\">4&nbsp;&nbsp;</span>Run local instance of the bedhost API</a></span></li></ul></div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create a tutorial directory and download demo files \n",
    "We need create a directory where we'll store the bedbase pipelines and files to be processed. We'll also need to create an environment variable that points to the tutorial directory (we'll need this variable in section 3 of the tutorial). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $HOME"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir bedbase_tutorial\n",
    "cd bedbase_tutorial\n",
    "export BEDBASEtutorial=\"$HOME/bedbase_tutorial\"\n",
    "source ~/.bashrc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To download the files we'll need for this tutorial, we can easily do it with the following commands:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2020-03-19 16:57:03--  http://big.databio.org/example_data/bedbase_demo/bedbase_demo_files_justBED/bedbase_BEDfiles.tar.gz\n",
      "Resolving big.databio.org (big.databio.org)... 128.143.245.181\n",
      "Connecting to big.databio.org (big.databio.org)|128.143.245.181|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 60245813 (57M) [application/octet-stream]\n",
      "Saving to: ‘bedbase_BEDfiles.tar.gz’\n",
      "\n",
      "bedbase_BEDfiles.ta 100%[===================>]  57.45M   914KB/s    in 59s     \n",
      "\n",
      "2020-03-19 16:58:02 (993 KB/s) - ‘bedbase_BEDfiles.tar.gz’ saved [60245813/60245813]\n",
      "\n",
      "--2020-03-19 16:58:02--  http://big.databio.org/example_data/bedbase_demo/bedbase_demo_files_justBED/bedbase_demo_PEPs.tar.gz\n",
      "Resolving big.databio.org (big.databio.org)... 128.143.245.181\n",
      "Connecting to big.databio.org (big.databio.org)|128.143.245.181|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 1374 (1.3K) [application/octet-stream]\n",
      "Saving to: ‘bedbase_demo_PEPs.tar.gz’\n",
      "\n",
      "bedbase_demo_PEPs.t 100%[===================>]   1.34K  --.-KB/s    in 0s      \n",
      "\n",
      "2020-03-19 16:58:03 (88.6 MB/s) - ‘bedbase_demo_PEPs.tar.gz’ saved [1374/1374]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "wget http://big.databio.org/example_data/bedbase_demo/bedbase_demo_files_justBED/bedbase_BEDfiles.tar.gz     \n",
    "wget http://big.databio.org/example_data/bedbase_demo/bedbase_demo_files_justBED/bedbase_demo_PEPs.tar.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The downloaded files are compressed so we'll need to untar them:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bedbase_BEDfiles/\n",
      "bedbase_BEDfiles/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19.bed.gz\n",
      "bedbase_BEDfiles/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105587_ENCFF413ANK_peaks_hg19.bed.gz\n",
      "bedbase_BEDfiles/GSM2423312_ENCFF155HVK_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE91663_ENCFF316ASR_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSM2423313_ENCFF722AOG_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19.bed.gz\n",
      "bedbase_BEDfiles/GSM2827349_ENCFF196DNQ_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF634NTU_peaks_hg19.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF937CGY_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSM2827350_ENCFF928JXU_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_demo_PEPs/\n",
      "bedbase_demo_PEPs/bedstat_annotation_sheet.csv\n",
      "bedbase_demo_PEPs/bedbuncher_config.yaml\n",
      "bedbase_demo_PEPs/bedbase_configuration.yaml\n",
      "bedbase_demo_PEPs/bedbuncher_query.csv\n",
      "bedbase_demo_PEPs/bedstat_config.yaml\n"
     ]
    }
   ],
   "source": [
    "tar -zxvf bedbase_BEDfiles.tar.gz\n",
    "tar -zxvf bedbase_demo_PEPs.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate statistics and plots of BED files using BEDSTAT\n",
    "\n",
    "\n",
    "### Create a PEP describing the BED files to process\n",
    "\n",
    "In order to get started, we'll need a PEP [Portable Encapsulated project](https://pepkit.github.io/). A PEP consists of 1) an annotation sheet (.csv) that contains information about the samples on a project and 2) a project config.yaml file that points to the sample annotation sheet. The config file also has other components, such as derived attributes, that in this case point to the BED files to be processed. The following is an example of a config file using the derived attributes `output_file_path` and `yaml_file` to point to the `.bed.gz` files and their respective metadata."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metadata:\n",
      "  sample_table: bedstat_annotation_sheet.csv\n",
      "  output_dir: ../bedstat/bedstat_pipeline_logs \n",
      "  pipeline_interfaces: ../bedstat/pipeline_interface.yaml\n",
      "\n",
      "constant_attributes: \n",
      "  output_file_path: \"source\"\n",
      "  yaml_file: \"source2\"\n",
      "  protocol: \"bedstat\"\n",
      "\n",
      "derived_attributes: [output_file_path, yaml_file]\n",
      "data_sources:\n",
      "  source: ../bedbase_BEDfiles/{file_name} \n",
      "  source2: ../bedstat/bedstat_pipeline_logs/submission/{sample_name}.yaml\n"
     ]
    }
   ],
   "source": [
    "cat bedbase_demo_PEPs/bedstat_config.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Download bedstat and the Bedbase configuration manager (bbconf)\n",
    "\n",
    "[bedstat](https://github.com/databio/bedstat) is a [pypiper](http://code.databio.org/pypiper/) pipeline that generates statistics and plots of BED files. Additionally, [bedstat](https://github.com/databio/bedstat) relies in\n",
    "[bbconf](https://github.com/databio/bbconf), the `bedbase` configuration manager which implements convenience methods for interacting with an elasticsearch database, where our files metadata will be placed. For carrying out this demo, we'll be using the dev version of `bbconf` that can be downloaded as follows:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'bedstat'...\n",
      "remote: Enumerating objects: 165, done.\u001b[K\n",
      "remote: Counting objects: 100% (165/165), done.\u001b[K\n",
      "remote: Compressing objects: 100% (92/92), done.\u001b[K\n",
      "remote: Total 362 (delta 81), reused 106 (delta 43), pack-reused 197\u001b[K\n",
      "Receiving objects: 100% (362/362), 57.94 KiB | 1.81 MiB/s, done.\n",
      "Resolving deltas: 100% (155/155), done.\n",
      "Cloning into 'bbconf'...\n",
      "remote: Enumerating objects: 251, done.\u001b[K\n",
      "remote: Counting objects: 100% (251/251), done.\u001b[K\n",
      "remote: Compressing objects: 100% (178/178), done.\u001b[K\n",
      "remote: Total 251 (delta 148), reused 154 (delta 61), pack-reused 0\u001b[K\n",
      "Receiving objects: 100% (251/251), 42.52 KiB | 4.72 MiB/s, done.\n",
      "Resolving deltas: 100% (148/148), done.\n",
      "Requirement already satisfied: piper in /home/jev4xy/.local/lib/python3.6/site-packages (0.12.1)\n",
      "Requirement already satisfied: psutil in /home/jev4xy/.local/lib/python3.6/site-packages (from piper) (5.6.5)\n",
      "Requirement already satisfied: logmuse>=0.2.4 in /home/jev4xy/.local/lib/python3.6/site-packages (from piper) (0.2.5)\n",
      "Requirement already satisfied: pandas in /home/jev4xy/.local/lib/python3.6/site-packages (from piper) (0.25.1)\n",
      "Requirement already satisfied: yacman in /home/jev4xy/.local/lib/python3.6/site-packages (from piper) (0.6.6)\n",
      "Requirement already satisfied: ubiquerg>=0.4.5 in /home/jev4xy/.local/lib/python3.6/site-packages (from piper) (0.5.0)\n",
      "Requirement already satisfied: attmap>=0.12.5 in /home/jev4xy/.local/lib/python3.6/site-packages (from piper) (0.12.9)\n",
      "Requirement already satisfied: pytz>=2017.2 in /home/jev4xy/.local/lib/python3.6/site-packages (from pandas->piper) (2019.2)\n",
      "Requirement already satisfied: numpy>=1.13.3 in /home/jev4xy/.local/lib/python3.6/site-packages (from pandas->piper) (1.17.2)\n",
      "Requirement already satisfied: python-dateutil>=2.6.1 in /home/jev4xy/.local/lib/python3.6/site-packages (from pandas->piper) (2.8.0)\n",
      "Requirement already satisfied: pyyaml>=3.13 in /home/jev4xy/.local/lib/python3.6/site-packages (from yacman->piper) (5.1.2)\n",
      "Requirement already satisfied: oyaml in /home/jev4xy/.local/lib/python3.6/site-packages (from yacman->piper) (0.9)\n",
      "Requirement already satisfied: six>=1.5 in /home/jev4xy/.local/lib/python3.6/site-packages (from python-dateutil>=2.6.1->pandas->piper) (1.12.0)\n",
      "\u001b[33mWARNING: You are using pip version 19.3.1; however, version 20.0.2 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n",
      "Requirement already satisfied: loopercli in /home/jev4xy/.local/lib/python3.6/site-packages (0.12.4)\n",
      "Requirement already satisfied: attmap>=0.12.7 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.12.9)\n",
      "Requirement already satisfied: colorama>=0.3.9 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.4.1)\n",
      "Requirement already satisfied: divvy>=0.3.1 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.4.0)\n",
      "Requirement already satisfied: jinja2 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (2.10.1)\n",
      "Requirement already satisfied: logmuse>=0.2.0 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.2.5)\n",
      "Requirement already satisfied: ngstk>=0.0.1rc1 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.0.1rc2)\n",
      "Requirement already satisfied: pandas>=0.20.2 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.25.1)\n",
      "Requirement already satisfied: peppy>=0.22.2 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.22.4.dev0)\n",
      "Requirement already satisfied: pyyaml>=3.12 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (5.1.2)\n",
      "Requirement already satisfied: ubiquerg>=0.4.5 in /home/jev4xy/.local/lib/python3.6/site-packages (from loopercli) (0.5.0)\n",
      "Requirement already satisfied: yacman>=0.5.0 in /home/jev4xy/.local/lib/python3.6/site-packages (from divvy>=0.3.1->loopercli) (0.6.6)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /home/jev4xy/.local/lib/python3.6/site-packages (from jinja2->loopercli) (1.1.1)\n",
      "Requirement already satisfied: python-dateutil>=2.6.1 in /home/jev4xy/.local/lib/python3.6/site-packages (from pandas>=0.20.2->loopercli) (2.8.0)\n",
      "Requirement already satisfied: pytz>=2017.2 in /home/jev4xy/.local/lib/python3.6/site-packages (from pandas>=0.20.2->loopercli) (2019.2)\n",
      "Requirement already satisfied: numpy>=1.13.3 in /home/jev4xy/.local/lib/python3.6/site-packages (from pandas>=0.20.2->loopercli) (1.17.2)\n",
      "Requirement already satisfied: oyaml in /home/jev4xy/.local/lib/python3.6/site-packages (from yacman>=0.5.0->divvy>=0.3.1->loopercli) (0.9)\n",
      "Requirement already satisfied: six>=1.5 in /home/jev4xy/.local/lib/python3.6/site-packages (from python-dateutil>=2.6.1->pandas>=0.20.2->loopercli) (1.12.0)\n",
      "\u001b[33mWARNING: You are using pip version 19.3.1; however, version 20.0.2 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "git clone git@github.com:databio/bedstat\n",
    "git clone -b dev git@github.com:databio/bbconf\n",
    "\n",
    "# Install Python dependencies\n",
    "pip install piper --user\n",
    "#pip install git+https://github.com/databio/bbconf.git@dev --user\n",
    "\n",
    "# Install R dependencies\n",
    "Rscript scripts/installRdeps.R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We'll need to create a directory where we can store the stats and plots generated by `bedstat`. Additionally, we'll create a directory where we can store log and metadata files that we'll need later on."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir bedstat/bedstat_output\n",
    "mkdir bedstat/bedstat_pipeline_logs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to use bbconf, we'll need to create a minimal configuration.yaml file. The path to this configuration file can be stored in the environment variable `$BEDBASE`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "path:\n",
      "  pipelines_output: ../bedstat/bedstat_output\n",
      "\n",
      "database:\n",
      "  host: localhost\n",
      "  bed_index: bed_index\n",
      "  bedset_index: bedset_index\n",
      "\n",
      "server:\n",
      "  host: 0.0.0.0\n",
      "  port: 8000\n"
     ]
    }
   ],
   "source": [
    "cat bedbase_demo_PEPs/bedbase_configuration.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Inititiate a local elasticsearch cluster\n",
    "\n",
    "In addition to generate statistics and plots, [bedstat](https://github.com/databio/bedstat) inserts JSON formatted metadata into an [elasticsearch](https://www.elastic.co/elasticsearch/?ultron=[EL]-[B]-[AMER]-US+CA-Exact&blade=adwords-s&Device=c&thor=elasticsearch&gclid=Cj0KCQjwjcfzBRCHARIsAO-1_Oq5mSdze16kripxT5_I__EeH9F-xUCz_khEvzGL7q_mqP62CahJ9SIaAg2BEALw_wcB) database that it'll later be used to search and extract files and information about them. (This step may have to be performed outside the notebook since these commands ask for a sudo password. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[sudo] password for jev4xy: \r\n"
     ]
    }
   ],
   "source": [
    "# If docker is not already installed, you can do so with the following commands\n",
    "#(make sure you have sudo permissions)\n",
    "\n",
    "sudo apt-get update\n",
    "sudo apt-get install docker-engine -y\n",
    "\n",
    "# Create a persistent volume to house elastic search data\n",
    "sudo docker volume create es-data\n",
    "\n",
    "# Run the docker container for elasticsearch\n",
    "sudo docker run -p 9200:9200 -p 9300:9300 -v es-data:/usr/share/elasticsearch/data -e \"xpack.ml.enabled=false\" \\\n",
    "  -e \"discovery.type=single-node\" elasticsearch:7.5.1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run bedstat  on the demo PEP\n",
    "To run [bedstat](https://github.com/databio/bedstat) and the other required pipelines in this demo, we will rely on the pipeline submission engine [looper](http://looper.databio.org/en/latest/),which can be installed in the following manner"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install --user loopercli"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In order to establish a modular connection between a project and a pipeline, we'll need to create a [pipeline interface](http://looper.databio.org/en/latest/linking-a-pipeline/) file, which tells looper how to run the pipeline. If `bedstat` is being run from an HPC environment where docker is not available, we recommend running the pipeline using the `--no-db-commit` flag (this will only calculate statistics and generate plots but will not insert this information into the local elasticsearch cluster. Once we have generated plots and statistics, we can insert them into our local elasticsearch cluster running `bedstat` with the `--just-db-commit` flag. If your data lives on a local environment, as it's the case in this tutorial, it's not necessary to set those flags and we can run bedstat in the following manner:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command: run (Looper version: 0.12.4)\n",
      "Reading sample table: '/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/bedstat_annotation_sheet.csv'\n",
      "Activating compute package 'local'\n",
      "Finding pipelines for protocol(s): bedstat\n",
      "Known protocols: bedstat\n",
      "'/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py' appears to attempt to run on import; does it lack a conditional on '__main__'? Using base type: Sample\n",
      "\u001b[36m## [1 of 15] bedbase_demo_db1 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db1.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db1.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:38:38\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db1.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/\n",
      "*  Pipeline started at:   (03-19 17:38:38) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db1.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz --fileId=GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984 --genome=hg38 --digest=78c0e4753d04b238fc07e4ebe5a02984` (13472)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 11017\n",
      "promoterProx :\tfound 2911\n",
      "exon :\tfound 11031\n",
      "intron :\tfound 22553\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:17. Running peak memory: 0.611GB.  \n",
      "  PID: 13472;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.611GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:17\n",
      "*  Total elapsed time (all runs):  0:00:17\n",
      "*         Peak memory (this run):  0.611 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:38:56\n",
      "\u001b[36m## [2 of 15] bedbase_demo_db2 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db2.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db2.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:38:56\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db2.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/fdd94ac0787599d564b07193e4ec41fd/\n",
      "*  Pipeline started at:   (03-19 17:38:56) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db2.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/fdd94ac0787599d564b07193e4ec41fd/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38.bed.gz --fileId=GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/fdd94ac0787599d564b07193e4ec41fd --genome=hg38 --digest=fdd94ac0787599d564b07193e4ec41fd` (13528)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/fdd94ac0787599d564b07193e4ec41fd/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/fdd94ac0787599d564b07193e4ec41fd/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/fdd94ac0787599d564b07193e4ec41fd/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 102\n",
      "promoterProx :\tfound 178\n",
      "exon :\tfound 334\n",
      "intron :\tfound 2938\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/fdd94ac0787599d564b07193e4ec41fd/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chrUn_KI270466v1, chrUn_KI270467v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CT [... truncated]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chrUn_KI270466v1, chrUn_KI270467v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CT [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chrUn_KI270466v1, chrUn_KI270467v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CT [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chrUn_KI270466v1, chrUn_KI270467v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CT [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:12. Running peak memory: 0.385GB.  \n",
      "  PID: 13528;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.385GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:12\n",
      "*  Total elapsed time (all runs):  0:00:12\n",
      "*         Peak memory (this run):  0.385 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:39:09\n",
      "\u001b[36m## [3 of 15] bedbase_demo_db3 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db3.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db3.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:39:09\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db3.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a6a08126cb6f4b1953ba0ec8675df85a/\n",
      "*  Pipeline started at:   (03-19 17:39:09) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db3.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a6a08126cb6f4b1953ba0ec8675df85a/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38.bed.gz --fileId=GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a6a08126cb6f4b1953ba0ec8675df85a --genome=hg38 --digest=a6a08126cb6f4b1953ba0ec8675df85a` (13577)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a6a08126cb6f4b1953ba0ec8675df85a/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a6a08126cb6f4b1953ba0ec8675df85a/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a6a08126cb6f4b1953ba0ec8675df85a/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 31\n",
      "promoterProx :\tfound 59\n",
      "exon :\tfound 156\n",
      "intron :\tfound 1595\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a6a08126cb6f4b1953ba0ec8675df85a/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270587v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CTG7, chrCHR_HSCHR11_1_CTG8, chrCHR_HS [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270587v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CTG7, chrCHR_HSCHR11_1_CTG8, chrCHR_HS [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270587v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CTG7, chrCHR_HSCHR11_1_CTG8, chrCHR_HS [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270587v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrCHR_HSCHR11_1_CTG5, chrCHR_HSCHR11_1_CTG6, chrCHR_HSCHR11_1_CTG7, chrCHR_HSCHR11_1_CTG8, chrCHR_HS [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:11. Running peak memory: 0.385GB.  \n",
      "  PID: 13577;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.385GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:11\n",
      "*  Total elapsed time (all runs):  0:00:11\n",
      "*         Peak memory (this run):  0.3851 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:39:20\n",
      "\u001b[36m## [4 of 15] bedbase_demo_db4 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db4.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db4.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:39:21\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105977_ENCFF937CGY_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db4.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a78493a2b314afe9f6635c4883f0d44b/\n",
      "*  Pipeline started at:   (03-19 17:39:21) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105977_ENCFF937CGY_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db4.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a78493a2b314afe9f6635c4883f0d44b/GSE105977_ENCFF937CGY_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105977_ENCFF937CGY_peaks_GRCh38.bed.gz --fileId=GSE105977_ENCFF937CGY_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a78493a2b314afe9f6635c4883f0d44b --genome=hg38 --digest=a78493a2b314afe9f6635c4883f0d44b` (13632)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a78493a2b314afe9f6635c4883f0d44b/GSE105977_ENCFF937CGY_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a78493a2b314afe9f6635c4883f0d44b/GSE105977_ENCFF937CGY_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a78493a2b314afe9f6635c4883f0d44b/GSE105977_ENCFF937CGY_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 1748\n",
      "promoterProx :\tfound 6890\n",
      "exon :\tfound 20173\n",
      "intron :\tfound 137782\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a78493a2b314afe9f6635c4883f0d44b/GSE105977_ENCFF937CGY_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270466v1, chrUn_KI270467v1, chr1_KI270711v1_random, chrUn_GL000224v1, chrUn_KI270448v1, chrUn_KI270587v1, chr1_KI270713v1_random, chrUn_KI270511v1, chrUn_KI270538v1, chr14_GL000225v1_random, chr5_GL000208v1_random, chrUn_GL000195v1, chrUn_GL000214v1, chr1_KI270712v1_random, chr22_KI270735v1_random, chr17_GL000205v2_random, chrUn_KI270742v1, chrUn_KI270757v1, chrUn_KI270438v1, chr2_KI270716v1_random, chrUn_KI270337v1, chr14_KI270722v1_random, chrUn_KI270519v1, chr1_KI270709v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_GL000216v2, chrUn_KI270310v1, chrUn_KI270746v1, chrUn_KI270749v1, chr14_GL000009v2_random, chrUn_GL000219v1, chr1_KI270708v1_random, chrUn_KI270507v1, chrUn_KI270508v1, chr22_KI270739v1_random, chrUn_KI270515v1, chr17_KI270729v1_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCH [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270466v1, chrUn_KI270467v1, chr1_KI270711v1_random, chrUn_GL000224v1, chrUn_KI270448v1, chrUn_KI270587v1, chr1_KI270713v1_random, chrUn_KI270511v1, chrUn_KI270538v1, chr14_GL000225v1_random, chr5_GL000208v1_random, chrUn_GL000195v1, chrUn_GL000214v1, chr1_KI270712v1_random, chr22_KI270735v1_random, chr17_GL000205v2_random, chrUn_KI270742v1, chrUn_KI270757v1, chrUn_KI270438v1, chr2_KI270716v1_random, chrUn_KI270337v1, chr14_KI270722v1_random, chrUn_KI270519v1, chr1_KI270709v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_GL000216v2, chrUn_KI270310v1, chrUn_KI270746v1, chrUn_KI270749v1, chr14_GL000009v2_random, chrUn_GL000219v1, chr1_KI270708v1_random, chrUn_KI270507v1, chrUn_KI270508v1, chr22_KI270739v1_random, chrUn_KI270515v1, chr17_KI270729v1_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCH [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270466v1, chrUn_KI270467v1, chr1_KI270711v1_random, chrUn_GL000224v1, chrUn_KI270448v1, chrUn_KI270587v1, chr1_KI270713v1_random, chrUn_KI270511v1, chrUn_KI270538v1, chr14_GL000225v1_random, chr5_GL000208v1_random, chrUn_GL000195v1, chrUn_GL000214v1, chr1_KI270712v1_random, chr22_KI270735v1_random, chr17_GL000205v2_random, chrUn_KI270742v1, chrUn_KI270757v1, chrUn_KI270438v1, chr2_KI270716v1_random, chrUn_KI270337v1, chr14_KI270722v1_random, chrUn_KI270519v1, chr1_KI270709v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_GL000216v2, chrUn_KI270310v1, chrUn_KI270746v1, chrUn_KI270749v1, chr14_GL000009v2_random, chrUn_GL000219v1, chr1_KI270708v1_random, chrUn_KI270507v1, chrUn_KI270508v1, chr22_KI270739v1_random, chrUn_KI270515v1, chr17_KI270729v1_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCH [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_KI270466v1, chrUn_KI270467v1, chr1_KI270711v1_random, chrUn_GL000224v1, chrUn_KI270448v1, chrUn_KI270587v1, chr1_KI270713v1_random, chrUn_KI270511v1, chrUn_KI270538v1, chr14_GL000225v1_random, chr5_GL000208v1_random, chrUn_GL000195v1, chrUn_GL000214v1, chr1_KI270712v1_random, chr22_KI270735v1_random, chr17_GL000205v2_random, chrUn_KI270742v1, chrUn_KI270757v1, chrUn_KI270438v1, chr2_KI270716v1_random, chrUn_KI270337v1, chr14_KI270722v1_random, chrUn_KI270519v1, chr1_KI270709v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_GL000216v2, chrUn_KI270310v1, chrUn_KI270746v1, chrUn_KI270749v1, chr14_GL000009v2_random, chrUn_GL000219v1, chr1_KI270708v1_random, chrUn_KI270507v1, chrUn_KI270508v1, chr22_KI270739v1_random, chrUn_KI270515v1, chr17_KI270729v1_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCH [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:34. Running peak memory: 0.461GB.  \n",
      "  PID: 13632;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.461GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:34\n",
      "*  Total elapsed time (all runs):  0:00:34\n",
      "*         Peak memory (this run):  0.4613 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:39:55\n",
      "\u001b[36m## [5 of 15] bedbase_demo_db5 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db5.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db5.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:39:55\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE91663_ENCFF316ASR_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db5.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/50e19bd44174bb286aa28ae2a15e7b8f/\n",
      "*  Pipeline started at:   (03-19 17:39:55) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE91663_ENCFF316ASR_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db5.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/50e19bd44174bb286aa28ae2a15e7b8f/GSE91663_ENCFF316ASR_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE91663_ENCFF316ASR_peaks_GRCh38.bed.gz --fileId=GSE91663_ENCFF316ASR_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/50e19bd44174bb286aa28ae2a15e7b8f --genome=hg38 --digest=50e19bd44174bb286aa28ae2a15e7b8f` (13694)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/50e19bd44174bb286aa28ae2a15e7b8f/GSE91663_ENCFF316ASR_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/50e19bd44174bb286aa28ae2a15e7b8f/GSE91663_ENCFF316ASR_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/50e19bd44174bb286aa28ae2a15e7b8f/GSE91663_ENCFF316ASR_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 7468\n",
      "promoterProx :\tfound 13588\n",
      "exon :\tfound 30763\n",
      "intron :\tfound 127116\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/50e19bd44174bb286aa28ae2a15e7b8f/GSE91663_ENCFF316ASR_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_KI270744v1, chrUn_GL000219v1, chrUn_KI270742v1, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_GL000195v1, chr1_KI270707v1_random, chr1_KI270711v1_random, chr14_GL000009v2_random, chr14_GL000194v1_random, chrUn_GL000216v2, chr16_KI270728v1_random, chr9_KI270718v1_random, chr22_KI270736v1_random, chr4_GL000008v2_random, chrUn_GL000214v1, chrUn_KI270442v1, chrUn_KI270745v1, chr22_KI270738v1_random, chr22_KI270733v1_random, chrUn_KI270749v1, chrUn_KI270589v1, chr17_KI270729v1_random, chr22_KI270734v1_random, chr14_KI270725v1_random, chr5_GL000208v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG20 [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_KI270744v1, chrUn_GL000219v1, chrUn_KI270742v1, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_GL000195v1, chr1_KI270707v1_random, chr1_KI270711v1_random, chr14_GL000009v2_random, chr14_GL000194v1_random, chrUn_GL000216v2, chr16_KI270728v1_random, chr9_KI270718v1_random, chr22_KI270736v1_random, chr4_GL000008v2_random, chrUn_GL000214v1, chrUn_KI270442v1, chrUn_KI270745v1, chr22_KI270738v1_random, chr22_KI270733v1_random, chrUn_KI270749v1, chrUn_KI270589v1, chr17_KI270729v1_random, chr22_KI270734v1_random, chr14_KI270725v1_random, chr5_GL000208v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG20 [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_KI270744v1, chrUn_GL000219v1, chrUn_KI270742v1, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_GL000195v1, chr1_KI270707v1_random, chr1_KI270711v1_random, chr14_GL000009v2_random, chr14_GL000194v1_random, chrUn_GL000216v2, chr16_KI270728v1_random, chr9_KI270718v1_random, chr22_KI270736v1_random, chr4_GL000008v2_random, chrUn_GL000214v1, chrUn_KI270442v1, chrUn_KI270745v1, chr22_KI270738v1_random, chr22_KI270733v1_random, chrUn_KI270749v1, chrUn_KI270589v1, chr17_KI270729v1_random, chr22_KI270734v1_random, chr14_KI270725v1_random, chr5_GL000208v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG20 [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_KI270744v1, chrUn_GL000219v1, chrUn_KI270742v1, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_GL000195v1, chr1_KI270707v1_random, chr1_KI270711v1_random, chr14_GL000009v2_random, chr14_GL000194v1_random, chrUn_GL000216v2, chr16_KI270728v1_random, chr9_KI270718v1_random, chr22_KI270736v1_random, chr4_GL000008v2_random, chrUn_GL000214v1, chrUn_KI270442v1, chrUn_KI270745v1, chr22_KI270738v1_random, chr22_KI270733v1_random, chrUn_KI270749v1, chrUn_KI270589v1, chr17_KI270729v1_random, chr22_KI270734v1_random, chr14_KI270725v1_random, chr5_GL000208v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG20 [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:27. Running peak memory: 0.462GB.  \n",
      "  PID: 13694;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.462GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:27\n",
      "*  Total elapsed time (all runs):  0:00:27\n",
      "*         Peak memory (this run):  0.4618 GB\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*        Pipeline completed time: 2020-03-19 17:40:22\n",
      "\u001b[36m## [6 of 15] bedbase_demo_db6 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db6.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db6.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:40:22\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db6.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9cd65cf4f07b83af35770c4a098fd4c6/\n",
      "*  Pipeline started at:   (03-19 17:40:23) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db6.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9cd65cf4f07b83af35770c4a098fd4c6/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38.bed.gz --fileId=GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9cd65cf4f07b83af35770c4a098fd4c6 --genome=hg38 --digest=9cd65cf4f07b83af35770c4a098fd4c6` (13745)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9cd65cf4f07b83af35770c4a098fd4c6/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9cd65cf4f07b83af35770c4a098fd4c6/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9cd65cf4f07b83af35770c4a098fd4c6/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 1559\n",
      "promoterProx :\tfound 1208\n",
      "exon :\tfound 2416\n",
      "intron :\tfound 5794\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9cd65cf4f07b83af35770c4a098fd4c6/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr17_GL000205v2_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrC [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr17_GL000205v2_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrC [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr17_GL000205v2_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrC [... truncated]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr17_GL000205v2_random, chrUn_KI270744v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_1_CTG2, chrCHR_HSCHR10_1_CTG4, chrCHR_HSCHR11_1_CTG1_2, chrC [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:13. Running peak memory: 0.384GB.  \n",
      "  PID: 13745;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.384GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:13\n",
      "*  Total elapsed time (all runs):  0:00:13\n",
      "*         Peak memory (this run):  0.3837 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:40:35\n",
      "\u001b[36m## [7 of 15] bedbase_demo_db7 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db7.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db7.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:40:35\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db7.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a5af5857bfbc3bfc8fea09cb90e67a16/\n",
      "*  Pipeline started at:   (03-19 17:40:36) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db7.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a5af5857bfbc3bfc8fea09cb90e67a16/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38.bed.gz --fileId=GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a5af5857bfbc3bfc8fea09cb90e67a16 --genome=hg38 --digest=a5af5857bfbc3bfc8fea09cb90e67a16` (13804)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a5af5857bfbc3bfc8fea09cb90e67a16/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a5af5857bfbc3bfc8fea09cb90e67a16/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a5af5857bfbc3bfc8fea09cb90e67a16/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 1967\n",
      "promoterProx :\tfound 1607\n",
      "exon :\tfound 3285\n",
      "intron :\tfound 8071\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/a5af5857bfbc3bfc8fea09cb90e67a16/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000219v1, chr1_KI270711v1_random, chrUn_KI270744v1, chr1_KI270714v1_random, chr1_KI270713v1_random, chrUn_KI270742v1, chr1_KI270706v1_random, chr17_GL000205v2_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_ [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000219v1, chr1_KI270711v1_random, chrUn_KI270744v1, chr1_KI270714v1_random, chr1_KI270713v1_random, chrUn_KI270742v1, chr1_KI270706v1_random, chr17_GL000205v2_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_ [... truncated]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000219v1, chr1_KI270711v1_random, chrUn_KI270744v1, chr1_KI270714v1_random, chr1_KI270713v1_random, chrUn_KI270742v1, chr1_KI270706v1_random, chr17_GL000205v2_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_ [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000219v1, chr1_KI270711v1_random, chrUn_KI270744v1, chr1_KI270714v1_random, chr1_KI270713v1_random, chrUn_KI270742v1, chr1_KI270706v1_random, chr17_GL000205v2_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, chrCHR_HG986_PATCH, chrCHR_HSCHR10_1_CTG1, chrCHR_HSCHR10_ [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:14. Running peak memory: 0.879GB.  \n",
      "  PID: 13804;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.879GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:14\n",
      "*  Total elapsed time (all runs):  0:00:14\n",
      "*         Peak memory (this run):  0.8793 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:40:50\n",
      "\u001b[36m## [8 of 15] bedbase_demo_db8 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db8.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db8.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:40:50\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSM2423312_ENCFF155HVK_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db8.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/02fd518818560c28ed20ed98f4c291bd/\n",
      "*  Pipeline started at:   (03-19 17:40:50) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSM2423312_ENCFF155HVK_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db8.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/02fd518818560c28ed20ed98f4c291bd/GSM2423312_ENCFF155HVK_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSM2423312_ENCFF155HVK_peaks_GRCh38.bed.gz --fileId=GSM2423312_ENCFF155HVK_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/02fd518818560c28ed20ed98f4c291bd --genome=hg38 --digest=02fd518818560c28ed20ed98f4c291bd` (13883)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/02fd518818560c28ed20ed98f4c291bd/GSM2423312_ENCFF155HVK_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/02fd518818560c28ed20ed98f4c291bd/GSM2423312_ENCFF155HVK_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/02fd518818560c28ed20ed98f4c291bd/GSM2423312_ENCFF155HVK_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 6459\n",
      "promoterProx :\tfound 13322\n",
      "exon :\tfound 29119\n",
      "intron :\tfound 129565\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/02fd518818560c28ed20ed98f4c291bd/GSM2423312_ENCFF155HVK_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_KI270742v1, chrUn_KI270744v1, chr1_KI270711v1_random, chr1_KI270706v1_random, chr22_KI270731v1_random, chrUn_GL000195v1, chr14_GL000194v1_random, chrUn_KI270442v1, chr17_KI270729v1_random, chr1_KI270707v1_random, chr22_KI270736v1_random, chr1_KI270709v1_random, chr22_KI270733v1_random, chr4_GL000008v2_random, chr16_KI270728v1_random, chr9_KI270719v1_random, chr22_KI270732v1_random, chr14_GL000009v2_random, chrUn_KI270745v1, chr14_GL000225v1_random, chrUn_KI270330v1, chrUn_GL000220v1, chr22_KI270737v1_random, chrUn_KI270751v1, chrUn_KI270757v1, chr9_KI270720v1_random, chrUn_GL000216v2, chr14_KI270724v1_random, chrUn_KI270746v1, chr9_KI270718v1_random, chrUn_KI270587v1, chrUn_GL000214v1, chr1_KI270712v1_random\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chr [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_KI270742v1, chrUn_KI270744v1, chr1_KI270711v1_random, chr1_KI270706v1_random, chr22_KI270731v1_random, chrUn_GL000195v1, chr14_GL000194v1_random, chrUn_KI270442v1, chr17_KI270729v1_random, chr1_KI270707v1_random, chr22_KI270736v1_random, chr1_KI270709v1_random, chr22_KI270733v1_random, chr4_GL000008v2_random, chr16_KI270728v1_random, chr9_KI270719v1_random, chr22_KI270732v1_random, chr14_GL000009v2_random, chrUn_KI270745v1, chr14_GL000225v1_random, chrUn_KI270330v1, chrUn_GL000220v1, chr22_KI270737v1_random, chrUn_KI270751v1, chrUn_KI270757v1, chr9_KI270720v1_random, chrUn_GL000216v2, chr14_KI270724v1_random, chrUn_KI270746v1, chr9_KI270718v1_random, chrUn_KI270587v1, chrUn_GL000214v1, chr1_KI270712v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chr [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_KI270742v1, chrUn_KI270744v1, chr1_KI270711v1_random, chr1_KI270706v1_random, chr22_KI270731v1_random, chrUn_GL000195v1, chr14_GL000194v1_random, chrUn_KI270442v1, chr17_KI270729v1_random, chr1_KI270707v1_random, chr22_KI270736v1_random, chr1_KI270709v1_random, chr22_KI270733v1_random, chr4_GL000008v2_random, chr16_KI270728v1_random, chr9_KI270719v1_random, chr22_KI270732v1_random, chr14_GL000009v2_random, chrUn_KI270745v1, chr14_GL000225v1_random, chrUn_KI270330v1, chrUn_GL000220v1, chr22_KI270737v1_random, chrUn_KI270751v1, chrUn_KI270757v1, chr9_KI270720v1_random, chrUn_GL000216v2, chr14_KI270724v1_random, chrUn_KI270746v1, chr9_KI270718v1_random, chrUn_KI270587v1, chrUn_GL000214v1, chr1_KI270712v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chr [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chr1_KI270714v1_random, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_KI270742v1, chrUn_KI270744v1, chr1_KI270711v1_random, chr1_KI270706v1_random, chr22_KI270731v1_random, chrUn_GL000195v1, chr14_GL000194v1_random, chrUn_KI270442v1, chr17_KI270729v1_random, chr1_KI270707v1_random, chr22_KI270736v1_random, chr1_KI270709v1_random, chr22_KI270733v1_random, chr4_GL000008v2_random, chr16_KI270728v1_random, chr9_KI270719v1_random, chr22_KI270732v1_random, chr14_GL000009v2_random, chrUn_KI270745v1, chr14_GL000225v1_random, chrUn_KI270330v1, chrUn_GL000220v1, chr22_KI270737v1_random, chrUn_KI270751v1, chrUn_KI270757v1, chr9_KI270720v1_random, chrUn_GL000216v2, chr14_KI270724v1_random, chrUn_KI270746v1, chr9_KI270718v1_random, chrUn_KI270587v1, chrUn_GL000214v1, chr1_KI270712v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chr [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:27. Running peak memory: 0.461GB.  \n",
      "  PID: 13883;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.461GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:27\n",
      "*  Total elapsed time (all runs):  0:00:27\n",
      "*         Peak memory (this run):  0.4611 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:41:17\n",
      "\u001b[36m## [9 of 15] bedhost_demo_db9 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedhost_demo_db9.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedhost_demo_db9.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:41:17\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSM2423313_ENCFF722AOG_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedhost_demo_db9.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/33d4328fe4ff3a472edff81bf8f5d566/\n",
      "*  Pipeline started at:   (03-19 17:41:17) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSM2423313_ENCFF722AOG_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedhost_demo_db9.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/33d4328fe4ff3a472edff81bf8f5d566/GSM2423313_ENCFF722AOG_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSM2423313_ENCFF722AOG_peaks_GRCh38.bed.gz --fileId=GSM2423313_ENCFF722AOG_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/33d4328fe4ff3a472edff81bf8f5d566 --genome=hg38 --digest=33d4328fe4ff3a472edff81bf8f5d566` (13960)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/33d4328fe4ff3a472edff81bf8f5d566/GSM2423313_ENCFF722AOG_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/33d4328fe4ff3a472edff81bf8f5d566/GSM2423313_ENCFF722AOG_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/33d4328fe4ff3a472edff81bf8f5d566/GSM2423313_ENCFF722AOG_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 5514\n",
      "promoterProx :\tfound 12714\n",
      "exon :\tfound 28439\n",
      "intron :\tfound 128655\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/33d4328fe4ff3a472edff81bf8f5d566/GSM2423313_ENCFF722AOG_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chrUn_KI270744v1, chr17_GL000205v2_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_KI270742v1, chrUn_GL000216v2, chrUn_GL000219v1, chr1_KI270711v1_random, chr16_KI270728v1_random, chr14_GL000009v2_random, chr1_KI270707v1_random, chr22_KI270738v1_random, chr22_KI270736v1_random, chrUn_KI270442v1, chr9_KI270718v1_random, chrUn_KI270751v1, chr1_KI270709v1_random, chrUn_KI270589v1, chr14_KI270725v1_random, chr14_GL000194v1_random, chrUn_GL000195v1, chr9_KI270719v1_random, chrUn_KI270538v1, chrUn_KI270749v1, chr4_GL000008v2_random, chr17_KI270729v1_random, chrUn_KI270330v1, chrUn_KI270745v1, chrUn_KI270746v1, chrUn_GL000214v1, chr22_KI270733v1_random, chr22_KI270737v1_random, chrUn_KI270756v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chrUn_KI270744v1, chr17_GL000205v2_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_KI270742v1, chrUn_GL000216v2, chrUn_GL000219v1, chr1_KI270711v1_random, chr16_KI270728v1_random, chr14_GL000009v2_random, chr1_KI270707v1_random, chr22_KI270738v1_random, chr22_KI270736v1_random, chrUn_KI270442v1, chr9_KI270718v1_random, chrUn_KI270751v1, chr1_KI270709v1_random, chrUn_KI270589v1, chr14_KI270725v1_random, chr14_GL000194v1_random, chrUn_GL000195v1, chr9_KI270719v1_random, chrUn_KI270538v1, chrUn_KI270749v1, chr4_GL000008v2_random, chr17_KI270729v1_random, chrUn_KI270330v1, chrUn_KI270745v1, chrUn_KI270746v1, chrUn_GL000214v1, chr22_KI270733v1_random, chr22_KI270737v1_random, chrUn_KI270756v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chrUn_KI270744v1, chr17_GL000205v2_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_KI270742v1, chrUn_GL000216v2, chrUn_GL000219v1, chr1_KI270711v1_random, chr16_KI270728v1_random, chr14_GL000009v2_random, chr1_KI270707v1_random, chr22_KI270738v1_random, chr22_KI270736v1_random, chrUn_KI270442v1, chr9_KI270718v1_random, chrUn_KI270751v1, chr1_KI270709v1_random, chrUn_KI270589v1, chr14_KI270725v1_random, chr14_GL000194v1_random, chrUn_GL000195v1, chr9_KI270719v1_random, chrUn_KI270538v1, chrUn_KI270749v1, chr4_GL000008v2_random, chr17_KI270729v1_random, chrUn_KI270330v1, chrUn_KI270745v1, chrUn_KI270746v1, chrUn_GL000214v1, chr22_KI270733v1_random, chr22_KI270737v1_random, chrUn_KI270756v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr1_KI270713v1_random, chrUn_KI270744v1, chr17_GL000205v2_random, chr1_KI270714v1_random, chr1_KI270706v1_random, chr14_GL000225v1_random, chrUn_KI270742v1, chrUn_GL000216v2, chrUn_GL000219v1, chr1_KI270711v1_random, chr16_KI270728v1_random, chr14_GL000009v2_random, chr1_KI270707v1_random, chr22_KI270738v1_random, chr22_KI270736v1_random, chrUn_KI270442v1, chr9_KI270718v1_random, chrUn_KI270751v1, chr1_KI270709v1_random, chrUn_KI270589v1, chr14_KI270725v1_random, chr14_GL000194v1_random, chrUn_GL000195v1, chr9_KI270719v1_random, chrUn_KI270538v1, chrUn_KI270749v1, chr4_GL000008v2_random, chr17_KI270729v1_random, chrUn_KI270330v1, chrUn_KI270745v1, chrUn_KI270746v1, chrUn_GL000214v1, chr22_KI270733v1_random, chr22_KI270737v1_random, chrUn_KI270756v1\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:26. Running peak memory: 0.462GB.  \n",
      "  PID: 13960;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.462GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:26\n",
      "*  Total elapsed time (all runs):  0:00:26\n",
      "*         Peak memory (this run):  0.4615 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:41:43\n",
      "\u001b[36m## [10 of 15] bedbase_demo_db10 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db10.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db10.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:41:44\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSM2827349_ENCFF196DNQ_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db10.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/2ffb2cedd14f5f1fae7cb765a66d82a3/\n",
      "*  Pipeline started at:   (03-19 17:41:44) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSM2827349_ENCFF196DNQ_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db10.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/2ffb2cedd14f5f1fae7cb765a66d82a3/GSM2827349_ENCFF196DNQ_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSM2827349_ENCFF196DNQ_peaks_GRCh38.bed.gz --fileId=GSM2827349_ENCFF196DNQ_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/2ffb2cedd14f5f1fae7cb765a66d82a3 --genome=hg38 --digest=2ffb2cedd14f5f1fae7cb765a66d82a3` (14015)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/2ffb2cedd14f5f1fae7cb765a66d82a3/GSM2827349_ENCFF196DNQ_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/2ffb2cedd14f5f1fae7cb765a66d82a3/GSM2827349_ENCFF196DNQ_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/2ffb2cedd14f5f1fae7cb765a66d82a3/GSM2827349_ENCFF196DNQ_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 13451\n",
      "promoterProx :\tfound 9142\n",
      "exon :\tfound 38026\n",
      "intron :\tfound 121299\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/2ffb2cedd14f5f1fae7cb765a66d82a3/GSM2827349_ENCFF196DNQ_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr17_GL000205v2_random, chrUn_GL000195v1, chr1_KI270713v1_random, chrUn_GL000220v1, chrUn_GL000224v1, chrUn_GL000219v1, chrUn_GL000218v1, chr17_KI270729v1_random, chr22_KI270733v1_random, chrUn_GL000216v2, chr1_KI270706v1_random, chr1_KI270714v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_KI270438v1, chr14_GL000225v1_random, chr16_KI270728v1_random, chr22_KI270736v1_random, chrUn_KI270515v1, chrUn_KI270517v1, chr22_KI270732v1_random, chrUn_KI270519v1, chr9_KI270719v1_random, chr14_KI270722v1_random, chrUn_KI270742v1, chr1_KI270709v1_random, chr2_KI270716v1_random, chr1_KI270712v1_random, chr1_KI270708v1_random, chrUn_KI270508v1, chrUn_KI270507v1, chrUn_KI270538v1, chr17_KI270730v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PA [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr17_GL000205v2_random, chrUn_GL000195v1, chr1_KI270713v1_random, chrUn_GL000220v1, chrUn_GL000224v1, chrUn_GL000219v1, chrUn_GL000218v1, chr17_KI270729v1_random, chr22_KI270733v1_random, chrUn_GL000216v2, chr1_KI270706v1_random, chr1_KI270714v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_KI270438v1, chr14_GL000225v1_random, chr16_KI270728v1_random, chr22_KI270736v1_random, chrUn_KI270515v1, chrUn_KI270517v1, chr22_KI270732v1_random, chrUn_KI270519v1, chr9_KI270719v1_random, chr14_KI270722v1_random, chrUn_KI270742v1, chr1_KI270709v1_random, chr2_KI270716v1_random, chr1_KI270712v1_random, chr1_KI270708v1_random, chrUn_KI270508v1, chrUn_KI270507v1, chrUn_KI270538v1, chr17_KI270730v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PA [... truncated]\n",
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr17_GL000205v2_random, chrUn_GL000195v1, chr1_KI270713v1_random, chrUn_GL000220v1, chrUn_GL000224v1, chrUn_GL000219v1, chrUn_GL000218v1, chr17_KI270729v1_random, chr22_KI270733v1_random, chrUn_GL000216v2, chr1_KI270706v1_random, chr1_KI270714v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_KI270438v1, chr14_GL000225v1_random, chr16_KI270728v1_random, chr22_KI270736v1_random, chrUn_KI270515v1, chrUn_KI270517v1, chr22_KI270732v1_random, chrUn_KI270519v1, chr9_KI270719v1_random, chr14_KI270722v1_random, chrUn_KI270742v1, chr1_KI270709v1_random, chr2_KI270716v1_random, chr1_KI270712v1_random, chr1_KI270708v1_random, chrUn_KI270508v1, chrUn_KI270507v1, chrUn_KI270538v1, chr17_KI270730v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PA [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chr17_GL000205v2_random, chrUn_GL000195v1, chr1_KI270713v1_random, chrUn_GL000220v1, chrUn_GL000224v1, chrUn_GL000219v1, chrUn_GL000218v1, chr17_KI270729v1_random, chr22_KI270733v1_random, chrUn_GL000216v2, chr1_KI270706v1_random, chr1_KI270714v1_random, chrUn_KI270751v1, chrUn_KI270442v1, chrUn_KI270438v1, chr14_GL000225v1_random, chr16_KI270728v1_random, chr22_KI270736v1_random, chrUn_KI270515v1, chrUn_KI270517v1, chr22_KI270732v1_random, chrUn_KI270519v1, chr9_KI270719v1_random, chr14_KI270722v1_random, chrUn_KI270742v1, chr1_KI270709v1_random, chr2_KI270716v1_random, chr1_KI270712v1_random, chr1_KI270708v1_random, chrUn_KI270508v1, chrUn_KI270507v1, chrUn_KI270538v1, chr17_KI270730v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PA [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:27. Running peak memory: 0.462GB.  \n",
      "  PID: 14015;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.462GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:27\n",
      "*  Total elapsed time (all runs):  0:00:27\n",
      "*         Peak memory (this run):  0.4622 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:42:11\n",
      "\u001b[36m## [11 of 15] bedbase_demo_db11 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db11.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db11.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:42:11\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSM2827350_ENCFF928JXU_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db11.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/3e67ac88348d8b816a8ca50ab94eeade/\n",
      "*  Pipeline started at:   (03-19 17:42:11) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSM2827350_ENCFF928JXU_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db11.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/3e67ac88348d8b816a8ca50ab94eeade/GSM2827350_ENCFF928JXU_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSM2827350_ENCFF928JXU_peaks_GRCh38.bed.gz --fileId=GSM2827350_ENCFF928JXU_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/3e67ac88348d8b816a8ca50ab94eeade --genome=hg38 --digest=3e67ac88348d8b816a8ca50ab94eeade` (14070)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/3e67ac88348d8b816a8ca50ab94eeade/GSM2827350_ENCFF928JXU_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/3e67ac88348d8b816a8ca50ab94eeade/GSM2827350_ENCFF928JXU_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "Error in GenomeInfoDb:::getDanglingSeqlevels(x, new2old = new2old, pruning.mode = pruning.mode,  : \n",
      "  The following seqlevels are to be dropped but are currently in use\n",
      "  (i.e. have ranges on them): chrEBV. Please use the 'pruning.mode'\n",
      "  argument to control how to prune 'x', that is, how to remove the ranges\n",
      "  in 'x' that are on these sequences. For example, do something like:\n",
      "  \n",
      "  seqlevels(x, pruning.mode=\"coarse\") <- new_seqlevels\n",
      "  \n",
      "  or\n",
      "  \n",
      "  keepSeqlevels(x, new_seqlevels, pruning.mode=\"coarse\")\n",
      "  \n",
      "  See ?seqinfo for a description of the pruning modes.\n",
      "Calls: doitall ... seqlevels<- -> seqinfo<- -> seqinfo<- -> <Anonymous>\n",
      "In addition: Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrEBV\n",
      "  - in 'y': chrM, chr1_GL383518v1_alt, chr1_GL383519v1_alt, chr1_GL383520v2_alt, chr1_KI270759v1_alt, chr1_KI270760v1_alt, chr1_KI270761v1_alt, chr1_KI270762v1_alt, chr1_KI270763v1_alt, chr1_KI270764v1_alt, chr1_KI270765v1_alt, chr1_KI270766v1_alt, chr1_KI270892v1_alt, chr2_GL383521v1_alt, chr2_GL383522v1_alt, chr2_GL582966v2_alt, chr2_KI270767v1_alt, chr2_KI270768v1_alt, chr2_KI270769v1_alt, chr2_KI270770v1_alt, chr2_KI270771v1_alt, chr2_KI270772v1_alt, chr2_KI270773v1_alt, chr2_KI270774v1_alt, chr2_KI270775v1_alt, chr2_KI270776v1_alt, chr2_KI270893v1_alt, chr2_KI270894v1_alt, chr3_GL383526v1_alt, chr3_JH636055v2_alt, chr3_KI270777v1_alt, chr3_KI270778v1_alt, chr3_KI270779v1_alt, chr3_KI270780v1_alt, chr3_KI270781v1_alt, chr3_KI270782v1_alt, chr3_KI270783v1_alt, chr3_KI270784v1_alt, chr3_KI270895v1_alt, chr3_KI270924v1_alt, chr3_KI270934v1_alt, chr3_KI270935v1_alt, chr3_KI270936v1_alt, chr3_KI27093 [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrM, chr1_GL383518v1_alt, chr1_GL383519v1_alt, chr1_GL383520v2_alt, chr1_KI270759v1_alt, chr1_KI270760v1_alt, chr1_KI270761v1_alt, chr1_KI270762v1_alt, chr1_KI270763v1_alt, chr1_KI270764v1_alt, chr1_KI270765v1_alt, chr1_KI270766v1_alt, chr1_KI270892v1_alt, chr2_GL383521v1_alt, chr2_GL383522v1_alt, chr2_GL582966v2_alt, chr2_KI270767v1_alt, chr2_KI270768v1_alt, chr2_KI270769v1_alt, chr2_KI270770v1_alt, chr2_KI270771v1_alt, chr2_KI270772v1_alt, chr2_KI270773v1_alt, chr2_KI270774v1_alt, chr2_KI270775v1_alt, chr2_KI270776v1_alt, chr2_KI270893v1_alt, chr2_KI270894v1_alt, chr3_GL383526v1_alt, chr3_JH636055v2_alt, chr3_KI270777v1_alt, chr3_KI270778v1_alt, chr3_KI270779v1_alt, chr3_KI270780v1_alt, chr3_KI270781v1_alt, chr3_KI270782v1_alt, chr3_KI270783v1_alt, chr3_KI270784v1_alt, chr3_KI270895v1_alt, chr3_KI270924v1_alt, chr3_KI270934v1_alt, chr3_KI270935v1_alt, chr3_KI270936v1_alt, chr3_KI270937v1_alt, chr4_GL000 [... truncated]\n",
      "Execution halted\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:17. Running peak memory: 0.461GB.  \n",
      "  PID: 14070;\tCommand: Rscript;\tReturn code: 1;\tMemory used: 0.461GB\n",
      "\n",
      "\n",
      "### Pipeline failed at:  (03-19 17:42:28) elapsed: 17.0 _TIME_\n",
      "\n",
      "Total time: 0:00:17\n",
      "Failure reason: Subprocess returned nonzero result. Check above output for details\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py\", line 50, in <module>\n",
      "    pm.run(cmd=command, target=json_file_path)\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/pypiper/manager.py\", line 785, in run\n",
      "    self.callprint(cmd, shell, lock_file, nofail, container)  # Run command\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/pypiper/manager.py\", line 1028, in callprint\n",
      "    self._triage_error(SubprocessError(msg), nofail)\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/pypiper/manager.py\", line 2131, in _triage_error\n",
      "    self.fail_pipeline(e)\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/pypiper/manager.py\", line 1660, in fail_pipeline\n",
      "    raise exc\n",
      "pypiper.exceptions.SubprocessError: Subprocess returned nonzero result. Check above output for details\n",
      "\u001b[36m## [12 of 15] bedbase_demo_db12 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db12.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db12.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:42:28\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105587_ENCFF413ANK_peaks_hg19.bed.gz --genome hg19 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db12.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/f41e12ddd3b6c4ee6da2140d0feee1ea/\n",
      "*  Pipeline started at:   (03-19 17:42:28) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105587_ENCFF413ANK_peaks_hg19.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg19`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db12.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/f41e12ddd3b6c4ee6da2140d0feee1ea/GSE105587_ENCFF413ANK_peaks_hg19.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105587_ENCFF413ANK_peaks_hg19.bed.gz --fileId=GSE105587_ENCFF413ANK_peaks_hg19 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/f41e12ddd3b6c4ee6da2140d0feee1ea --genome=hg19 --digest=f41e12ddd3b6c4ee6da2140d0feee1ea` (14123)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/f41e12ddd3b6c4ee6da2140d0feee1ea/GSE105587_ENCFF413ANK_peaks_hg19_tssdist\"\n",
      "Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/f41e12ddd3b6c4ee6da2140d0feee1ea/GSE105587_ENCFF413ANK_peaks_hg19_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg19.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/f41e12ddd3b6c4ee6da2140d0feee1ea/GSE105587_ENCFF413ANK_peaks_hg19_gccontent\"\n",
      "promoterCore :\tfound 15183\n",
      "promoterProx :\tfound 9310\n",
      "exon :\tfound 38478\n",
      "intron :\tfound 120148\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/f41e12ddd3b6c4ee6da2140d0feee1ea/GSE105587_ENCFF413ANK_peaks_hg19_partitions\"\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:33. Running peak memory: 0.721GB.  \n",
      "  PID: 14123;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.721GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:33\n",
      "*  Total elapsed time (all runs):  0:00:33\n",
      "*         Peak memory (this run):  0.7212 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:43:01\n",
      "\u001b[36m## [13 of 15] bedbase_demo_db13 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db13.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db13.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:43:01\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19.bed.gz --genome hg19 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db13.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9dc6f420639e0a265f3f179b6b42713a/\n",
      "*  Pipeline started at:   (03-19 17:43:02) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg19`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db13.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9dc6f420639e0a265f3f179b6b42713a/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19.bed.gz --fileId=GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9dc6f420639e0a265f3f179b6b42713a --genome=hg19 --digest=9dc6f420639e0a265f3f179b6b42713a` (14182)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9dc6f420639e0a265f3f179b6b42713a/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19_tssdist\"\n",
      "Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9dc6f420639e0a265f3f179b6b42713a/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg19.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9dc6f420639e0a265f3f179b6b42713a/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19_gccontent\"\n",
      "promoterCore :\tfound 11005\n",
      "promoterProx :\tfound 2977\n",
      "exon :\tfound 10918\n",
      "intron :\tfound 22468\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/9dc6f420639e0a265f3f179b6b42713a/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19_partitions\"\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:14. Running peak memory: 0.94GB.  \n",
      "  PID: 14182;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.94GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:14\n",
      "*  Total elapsed time (all runs):  0:00:14\n",
      "*         Peak memory (this run):  0.9405 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:43:15\n",
      "\u001b[36m## [14 of 15] bedbase_demo_db14 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db14.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db14.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:43:15\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19.bed.gz --genome hg19 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db14.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/e577eb947b5c791b30df969f0564324b/\n",
      "*  Pipeline started at:   (03-19 17:43:16) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg19`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db14.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/e577eb947b5c791b30df969f0564324b/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19.bed.gz --fileId=GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/e577eb947b5c791b30df969f0564324b --genome=hg19 --digest=e577eb947b5c791b30df969f0564324b` (14238)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/e577eb947b5c791b30df969f0564324b/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19_tssdist\"\n",
      "Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/e577eb947b5c791b30df969f0564324b/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg19.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/e577eb947b5c791b30df969f0564324b/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19_gccontent\"\n",
      "promoterCore :\tfound 98\n",
      "promoterProx :\tfound 184\n",
      "exon :\tfound 314\n",
      "intron :\tfound 3012\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/e577eb947b5c791b30df969f0564324b/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19_partitions\"\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:10. Running peak memory: 0.378GB.  \n",
      "  PID: 14238;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.378GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:10\n",
      "*  Total elapsed time (all runs):  0:00:10\n",
      "*         Peak memory (this run):  0.378 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:43:25\n",
      "\u001b[36m## [15 of 15] bedbase_demo_db15 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db15.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db15.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:43:26\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py --bedfile bedbase_BEDfiles/GSE105977_ENCFF634NTU_peaks_hg19.bed.gz --genome hg19 --sample-yaml bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db15.yaml -O bedstat/bedstat_pipeline_logs/results_pipeline --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml --no-db-commit -R`\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/021461e2edc7b85315042c5ef80c7c03/\n",
      "*  Pipeline started at:   (03-19 17:43:26) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  bd90e7cbb5a8146fe95bce6c38548da519cb7602\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-18 10:30:43 -0400\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*            `bedfile`:  `bedbase_BEDfiles/GSE105977_ENCFF634NTU_peaks_hg19.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg19`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `bedstat/bedstat_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `bedstat/bedstat_pipeline_logs/submission/bedbase_demo_db15.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/021461e2edc7b85315042c5ef80c7c03/GSE105977_ENCFF634NTU_peaks_hg19.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedbase_tutorial/bedstat/tools/regionstat.R --bedfile=bedbase_BEDfiles/GSE105977_ENCFF634NTU_peaks_hg19.bed.gz --fileId=GSE105977_ENCFF634NTU_peaks_hg19 --outputfolder=/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/021461e2edc7b85315042c5ef80c7c03 --genome=hg19 --digest=021461e2edc7b85315042c5ef80c7c03` (14288)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/021461e2edc7b85315042c5ef80c7c03/GSE105977_ENCFF634NTU_peaks_hg19_tssdist\"\n",
      "Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/021461e2edc7b85315042c5ef80c7c03/GSE105977_ENCFF634NTU_peaks_hg19_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg19.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/021461e2edc7b85315042c5ef80c7c03/GSE105977_ENCFF634NTU_peaks_hg19_gccontent\"\n",
      "promoterCore :\tfound 1756\n",
      "promoterProx :\tfound 7025\n",
      "exon :\tfound 19611\n",
      "intron :\tfound 137727\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/021461e2edc7b85315042c5ef80c7c03/GSE105977_ENCFF634NTU_peaks_hg19_partitions\"\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:25. Running peak memory: 0.671GB.  \n",
      "  PID: 14288;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.671GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:25\n",
      "*  Total elapsed time (all runs):  0:00:25\n",
      "*         Peak memory (this run):  0.671 GB\n",
      "*        Pipeline completed time: 2020-03-19 17:43:50\n",
      "\n",
      "Looper finished\n",
      "Samples valid for job generation: 15 of 15\n",
      "Successful samples: 15 of 15\n",
      "Commands submitted: 15 of 15\n",
      "Jobs submitted: 15\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "#looper run bedbase_demo_PEPs/bedstat_config.yaml --no-db-commit --compute local --limit 1 -R\n",
    "\n",
    "looper run bedbase_demo_PEPs/bedstat_config.yaml --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml \\\n",
    "--no-db-commit --compute local -R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command: run (Looper version: 0.12.4)\n",
      "Reading sample table: '/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/bedstat_annotation_sheet.csv'\n",
      "Activating compute package 'local'\n",
      "Finding pipelines for protocol(s): bedstat\n",
      "Known protocols: bedstat\n",
      "'/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py' appears to attempt to run on import; does it lack a conditional on '__main__'? Using base type: Sample\n",
      "\u001b[36m## [1 of 15] bedbase_demo_db1 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db1.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db1.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:53\n",
      "Established connection with Elasticsearch: localhost\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38'], 'gc_content': [0.5364], 'regions_no': [68728], 'mean_absolute_TSS_dist': [51994518.6441], 'md5sum': ['78c0e4753d04b238fc07e4ebe5a02984'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz'], 'exon_frequency': [11031], 'exon_percentage': [0.1605], 'intergenic_frequency': [21216], 'intergenic_percentage': [0.3087], 'intron_frequency': [22553], 'intron_percentage': [0.3281], 'promoterCore_frequency': [11017], 'promoterCore_percentage': [0.1603], 'promoterProx_frequency': [2911], 'promoterProx_percentage': [0.0424], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['GM12878'], 'tissue': ['NA'], 'antibody': ['IKZF1'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['IKZF1 ChIP-seq on human GM12878']}\n",
      "\u001b[36m## [2 of 15] bedbase_demo_db2 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db2.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db2.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:53\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38'], 'gc_content': [0.4153], 'regions_no': [5725], 'mean_absolute_TSS_dist': [55778498.7442], 'md5sum': ['fdd94ac0787599d564b07193e4ec41fd'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38.bed.gz'], 'exon_frequency': [334], 'exon_percentage': [0.0583], 'intergenic_frequency': [2173], 'intergenic_percentage': [0.3796], 'intron_frequency': [2938], 'intron_percentage': [0.5132], 'promoterCore_frequency': [102], 'promoterCore_percentage': [0.0178], 'promoterProx_frequency': [178], 'promoterProx_percentage': [0.0311], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['HEK293'], 'antibody': ['GLI2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['HEK293 cell line stably expressing N-terminal tagged eGFP-GLI2 under the control of a CMV promoter']}\n",
      "\u001b[36m## [3 of 15] bedbase_demo_db3 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db3.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db3.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:54\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38'], 'gc_content': [0.4063], 'regions_no': [2982], 'mean_absolute_TSS_dist': [51400911.9678], 'md5sum': ['a6a08126cb6f4b1953ba0ec8675df85a'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38.bed.gz'], 'exon_frequency': [156], 'exon_percentage': [0.0523], 'intergenic_frequency': [1141], 'intergenic_percentage': [0.3826], 'intron_frequency': [1595], 'intron_percentage': [0.5349], 'promoterCore_frequency': [31], 'promoterCore_percentage': [0.0104], 'promoterProx_frequency': [59], 'promoterProx_percentage': [0.0198], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['HEK293'], 'antibody': ['GLI2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['HEK293 cell line stably expressing N-terminal tagged eGFP-GLI2 under the control of a CMV promoter']}\n",
      "\u001b[36m## [4 of 15] bedbase_demo_db4 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db4.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db4.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:54\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105977_ENCFF937CGY_peaks_GRCh38'], 'gc_content': [0.391], 'regions_no': [300000], 'mean_absolute_TSS_dist': [52114640.1946], 'md5sum': ['a78493a2b314afe9f6635c4883f0d44b'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105977_ENCFF937CGY_peaks_GRCh38.bed.gz'], 'exon_frequency': [20173], 'exon_percentage': [0.0672], 'intergenic_frequency': [133407], 'intergenic_percentage': [0.4447], 'intron_frequency': [137782], 'intron_percentage': [0.4593], 'promoterCore_frequency': [1748], 'promoterCore_percentage': [0.0058], 'promoterProx_frequency': [6890], 'promoterProx_percentage': [0.023], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['HEK293'], 'antibody': ['GLI2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['HEK293 cell line stably expressing N-terminal tagged eGFP-GLI2 under the control of a CMV promoter']}\n",
      "\u001b[36m## [5 of 15] bedbase_demo_db5 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db5.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db5.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:55\n",
      "Established connection with Elasticsearch: localhost\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE91663_ENCFF316ASR_peaks_GRCh38'], 'gc_content': [0.4546], 'regions_no': [300000], 'mean_absolute_TSS_dist': [54150666.8582], 'md5sum': ['50e19bd44174bb286aa28ae2a15e7b8f'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE91663_ENCFF316ASR_peaks_GRCh38.bed.gz'], 'exon_frequency': [30763], 'exon_percentage': [0.1025], 'intergenic_frequency': [121065], 'intergenic_percentage': [0.4036], 'intron_frequency': [127116], 'intron_percentage': [0.4237], 'promoterCore_frequency': [7468], 'promoterCore_percentage': [0.0249], 'promoterProx_frequency': [13588], 'promoterProx_percentage': [0.0453], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': [' K562'], 'tissue': ['Lung '], 'antibody': [' ZEB2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['ZEB2 ChIP-seq on human K562 (ENCODE)']}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[36m## [6 of 15] bedbase_demo_db6 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db6.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db6.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:55\n",
      "Established connection with Elasticsearch: localhost\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38'], 'gc_content': [0.507], 'regions_no': [17110], 'mean_absolute_TSS_dist': [51414986.6069], 'md5sum': ['9cd65cf4f07b83af35770c4a098fd4c6'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38.bed.gz'], 'exon_frequency': [2416], 'exon_percentage': [0.1412], 'intergenic_frequency': [6133], 'intergenic_percentage': [0.3584], 'intron_frequency': [5794], 'intron_percentage': [0.3386], 'promoterCore_frequency': [1559], 'promoterCore_percentage': [0.0911], 'promoterProx_frequency': [1208], 'promoterProx_percentage': [0.0706], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': [' K562'], 'tissue': [' Lung'], 'antibody': [' ZEB2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['ZEB2 ChIP-seq on human K562 (ENCODE)']}\n",
      "\u001b[36m## [7 of 15] bedbase_demo_db7 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db7.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db7.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:56\n",
      "Established connection with Elasticsearch: localhost\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38'], 'gc_content': [0.503], 'regions_no': [23256], 'mean_absolute_TSS_dist': [51610783.8628], 'md5sum': ['a5af5857bfbc3bfc8fea09cb90e67a16'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38.bed.gz'], 'exon_frequency': [3285], 'exon_percentage': [0.1413], 'intergenic_frequency': [8326], 'intergenic_percentage': [0.358], 'intron_frequency': [8071], 'intron_percentage': [0.3471], 'promoterCore_frequency': [1967], 'promoterCore_percentage': [0.0846], 'promoterProx_frequency': [1607], 'promoterProx_percentage': [0.0691], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': [' K562'], 'tissue': [' Lung'], 'antibody': [' ZEB2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['ZEB2 ChIP-seq on human K562 (ENCODE)']}\n",
      "\u001b[36m## [8 of 15] bedbase_demo_db8 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db8.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db8.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:56\n",
      "Established connection with Elasticsearch: localhost\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSM2423312_ENCFF155HVK_peaks_GRCh38'], 'gc_content': [0.4502], 'regions_no': [300000], 'mean_absolute_TSS_dist': [53195047.1617], 'md5sum': ['02fd518818560c28ed20ed98f4c291bd'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSM2423312_ENCFF155HVK_peaks_GRCh38.bed.gz'], 'exon_frequency': [29119], 'exon_percentage': [0.0971], 'intergenic_frequency': [121535], 'intergenic_percentage': [0.4051], 'intron_frequency': [129565], 'intron_percentage': [0.4319], 'promoterCore_frequency': [6459], 'promoterCore_percentage': [0.0215], 'promoterProx_frequency': [13322], 'promoterProx_percentage': [0.0444], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': [' K562'], 'tissue': ['Lung'], 'antibody': [' ZEB2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['K562 established from pleural effusion of 53 year old female with chronic myelogenous leukemia']}\n",
      "\u001b[36m## [9 of 15] bedhost_demo_db9 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedhost_demo_db9.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedhost_demo_db9.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:56\n",
      "Established connection with Elasticsearch: localhost\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSM2423313_ENCFF722AOG_peaks_GRCh38'], 'gc_content': [0.4459], 'regions_no': [300000], 'mean_absolute_TSS_dist': [52479401.3593], 'md5sum': ['33d4328fe4ff3a472edff81bf8f5d566'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSM2423313_ENCFF722AOG_peaks_GRCh38.bed.gz'], 'exon_frequency': [28439], 'exon_percentage': [0.0948], 'intergenic_frequency': [124678], 'intergenic_percentage': [0.4156], 'intron_frequency': [128655], 'intron_percentage': [0.4288], 'promoterCore_frequency': [5514], 'promoterCore_percentage': [0.0184], 'promoterProx_frequency': [12714], 'promoterProx_percentage': [0.0424], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': [' K562'], 'tissue': ['Lung'], 'antibody': [' ZEB2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['K562 established from pleural effusion of 53 year old female with chronic myelogenous leukemia']}\n",
      "\u001b[36m## [10 of 15] bedbase_demo_db10 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db10.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db10.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:57\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSM2827349_ENCFF196DNQ_peaks_GRCh38'], 'gc_content': [0.4812], 'regions_no': [300000], 'mean_absolute_TSS_dist': [52671790.3859], 'md5sum': ['2ffb2cedd14f5f1fae7cb765a66d82a3'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSM2827349_ENCFF196DNQ_peaks_GRCh38.bed.gz'], 'exon_frequency': [38026], 'exon_percentage': [0.1268], 'intergenic_frequency': [118082], 'intergenic_percentage': [0.3936], 'intron_frequency': [121299], 'intron_percentage': [0.4043], 'promoterCore_frequency': [13451], 'promoterCore_percentage': [0.0448], 'promoterProx_frequency': [9142], 'promoterProx_percentage': [0.0305], 'genome': ['hg38'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['GM12878'], 'antibody': ['IKZF1'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['IKZF1 ChIP-seq on human GM12878']}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[36m## [11 of 15] bedbase_demo_db11 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db11.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db11.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:57\n",
      "Established connection with Elasticsearch: localhost\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedstat/pipeline/bedstat.py\", line 59, in <module>\n",
      "    with open(json_file_path, 'r', encoding='utf-8') as f:\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/3e67ac88348d8b816a8ca50ab94eeade/GSM2827350_ENCFF928JXU_peaks_GRCh38.json'\n",
      "\u001b[36m## [12 of 15] bedbase_demo_db12 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db12.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db12.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:58\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105587_ENCFF413ANK_peaks_hg19'], 'gc_content': [0.4894], 'regions_no': [300000], 'mean_absolute_TSS_dist': [53096969.8565], 'md5sum': ['f41e12ddd3b6c4ee6da2140d0feee1ea'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105587_ENCFF413ANK_peaks_hg19.bed.gz'], 'exon_frequency': [38478], 'exon_percentage': [0.1283], 'intergenic_frequency': [116881], 'intergenic_percentage': [0.3896], 'intron_frequency': [120148], 'intron_percentage': [0.4005], 'promoterCore_frequency': [15183], 'promoterCore_percentage': [0.0506], 'promoterProx_frequency': [9310], 'promoterProx_percentage': [0.031], 'genome': ['hg19'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['GM12878'], 'antibody': ['IKZF1'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['IKZF1 ChIP-seq on human GM12878']}\n",
      "\u001b[36m## [13 of 15] bedbase_demo_db13 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db13.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db13.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:58\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19'], 'gc_content': [0.5364], 'regions_no': [68681], 'mean_absolute_TSS_dist': [52337816.7597], 'md5sum': ['9dc6f420639e0a265f3f179b6b42713a'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19.bed.gz'], 'exon_frequency': [10918], 'exon_percentage': [0.159], 'intergenic_frequency': [21313], 'intergenic_percentage': [0.3103], 'intron_frequency': [22468], 'intron_percentage': [0.3271], 'promoterCore_frequency': [11005], 'promoterCore_percentage': [0.1602], 'promoterProx_frequency': [2977], 'promoterProx_percentage': [0.0433], 'genome': ['hg19'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['GM12878'], 'antibody': ['IKZF1'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['IKZF1 ChIP-seq on human GM12878']}\n",
      "\u001b[36m## [14 of 15] bedbase_demo_db14 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db14.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db14.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:59\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19'], 'gc_content': [0.4142], 'regions_no': [5837], 'mean_absolute_TSS_dist': [51149954.7245], 'md5sum': ['e577eb947b5c791b30df969f0564324b'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19.bed.gz'], 'exon_frequency': [314], 'exon_percentage': [0.0538], 'intergenic_frequency': [2229], 'intergenic_percentage': [0.3819], 'intron_frequency': [3012], 'intron_percentage': [0.516], 'promoterCore_frequency': [98], 'promoterCore_percentage': [0.0168], 'promoterProx_frequency': [184], 'promoterProx_percentage': [0.0315], 'genome': ['hg19'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['HEK293'], 'antibody': ['GLI2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['HEK293 cell line stably expressing N-terminal tagged eGFP-GLI2 under the control of a CMV promoter']}\n",
      "\u001b[36m## [15 of 15] bedbase_demo_db15 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db15.sub\n",
      "Job script (n=1; 0.00 Gb): bedstat/bedstat_pipeline_logs/submission/bedstat_bedbase_demo_db15.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 17:51:59\n",
      "Established connection with Elasticsearch: localhost\n",
      "'tissue' metadata not available\n",
      "'id' metadata not available\n",
      "'md5sum' metadata not available\n",
      "'plots' metadata not available\n",
      "'bedfile_path' metadata not available\n",
      "Data: {'id': ['GSE105977_ENCFF634NTU_peaks_hg19'], 'gc_content': [0.3911], 'regions_no': [300000], 'mean_absolute_TSS_dist': [57236876.5477], 'md5sum': ['021461e2edc7b85315042c5ef80c7c03'], 'plots': [{'name': 'tssdist', 'caption': 'Region-TSS distance distribution'}, {'name': 'chrombins', 'caption': 'Regions distribution over chromosomes'}, {'name': 'gccontent', 'caption': 'GC content'}, {'name': 'partitions', 'caption': 'Regions distribution over genomic partitions'}], 'bedfile_path': ['bedbase_BEDfiles/GSE105977_ENCFF634NTU_peaks_hg19.bed.gz'], 'exon_frequency': [19611], 'exon_percentage': [0.0654], 'intergenic_frequency': [133881], 'intergenic_percentage': [0.4463], 'intron_frequency': [137727], 'intron_percentage': [0.4591], 'promoterCore_frequency': [1756], 'promoterCore_percentage': [0.0059], 'promoterProx_frequency': [7025], 'promoterProx_percentage': [0.0234], 'genome': ['hg19'], 'exp_protocol': ['ChiPseq'], 'cell_type': ['HEK293'], 'antibody': ['GLI2'], 'treatment': ['None'], 'data_source': ['GEO'], 'description': ['HEK293 cell line stably expressing N-terminal tagged eGFP-GLI2 under the control of a CMV promoter']}\n",
      "\n",
      "Looper finished\n",
      "Samples valid for job generation: 15 of 15\n",
      "Successful samples: 15 of 15\n",
      "Commands submitted: 15 of 15\n",
      "Jobs submitted: 15\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "#looper run bedbase_demo_PEPs/bedstat_config.yaml  --just-db-commit --compute local -R\n",
    "\n",
    "looper run bedbase_demo_EPsP/bedstat_config.yaml --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml \\\n",
    "--just-db-commit --compute local -R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After the previous steps have been executed, our BED files should be available for query on our local elastic search cluster. Files can be queried using the `bedbuncher` pipeline described in the below section. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create bedsets using BEDBUNCHER\n",
    "\n",
    "### Create a new PEP describing the bedset name and specific JSON query  \n",
    "[bedbuncher](https://github.com/databio/bedbuncher) is a pipeline designed to create bedsets (sets of BED files retrieved from bedbase), with their respective statistics and additional outputs such as a `PEP` and an `iGD` database. In order to run `bedbuncher`, we will need to design an additional PEP describing the query as well as attributes such as the name assigned to the newly created bedset. This configuration file should point to the `JSON` file describing the query to find files of interest. The configuration file should have the following structure:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample_name,bedset_name,JSONquery_name,bbconfig_name,JSONquery_path,output_folder_path\n",
      "bedset1,bedbase_demo_bedset,test_query,bedbase_configuration,source1,source2\n"
     ]
    }
   ],
   "source": [
    "cat bedbase_demo_PEPs/bedbuncher_query.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metadata:\n",
      "  sample_table: bedbuncher_query.csv\n",
      "  output_dir: bedbuncher/bedbuncher_pipeline_logs\n",
      "  pipeline_interfaces: ../bedbuncher/pipeline_interface.yaml \n",
      "\n",
      "derived_attributes: [JSONquery_path, bbconfig_path]\n",
      "data_sources:\n",
      "  source1: \"bedbuncher/tests/{JSONquery_name}.json\"\n",
      "  source2: \"bedbase_demo_PEPs/{bbconfig_name}.yaml\"\n",
      "constant_attributes:\n",
      "  protocol: \"bedbuncher\"\n"
     ]
    }
   ],
   "source": [
    "cat bedbase_demo_PEPs/bedbuncher_config.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###  Download bedbuncher  and install CML dependencies\n",
    "\n",
    "To download the `bedbuncher` pipeline, simply clone the repository from github. Though not required, we'll also create a directory where we can store the pipeline logs. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'bedbuncher'...\n",
      "remote: Enumerating objects: 39, done.\u001b[K\n",
      "remote: Counting objects: 100% (39/39), done.\u001b[K\n",
      "remote: Compressing objects: 100% (27/27), done.\u001b[K\n",
      "remote: Total 235 (delta 22), reused 26 (delta 12), pack-reused 196\u001b[K\n",
      "Receiving objects: 100% (235/235), 54.59 KiB | 1.71 MiB/s, done.\n",
      "Resolving deltas: 100% (130/130), done.\n"
     ]
    }
   ],
   "source": [
    "git clone git@github.com:databio/bedbuncher\n",
    "mkdir bedbuncher/bedbuncher_pipeline_logs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "One of the feats of `bedbuncher` includes [iGD](https://github.com/databio/iGD) database creation from the files in the bedset. `iGD` can be installed by cloning the repository from github, executing the make file to create the binary, and pointing the binary location with the `$PATH` environment variable. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'iGD'...\n",
      "remote: Enumerating objects: 367, done.\u001b[K\n",
      "remote: Counting objects: 100% (367/367), done.\u001b[K\n",
      "remote: Compressing objects: 100% (253/253), done.\u001b[K\n",
      "remote: Total 1149 (delta 230), reused 242 (delta 111), pack-reused 782\u001b[K\n",
      "Receiving objects: 100% (1149/1149), 18.57 MiB | 1.14 MiB/s, done.\n",
      "Resolving deltas: 100% (651/651), done.\n",
      "mkdir -p obj\n",
      "mkdir -p bin\n",
      "cc -c -g -O2 -lz -lm src/igd_base.c -o obj/igd_base.o \n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kget_fileinfo\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:228:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfgets\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfgets(buf, 1024, fp)\u001b[m\u001b[K;//head line\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:235:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfgets\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfgets(buf, 1024, fp)\u001b[m\u001b[K;   //header\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kget_igdinfo\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:257:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfread(&iGD->nbp, sizeof(int32_t), 1, fp)\u001b[m\u001b[K;\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:258:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfread(&iGD->gType, sizeof(int32_t), 1, fp)\u001b[m\u001b[K;\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:259:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfread(&iGD->nCtg, sizeof(int32_t), 1, fp)\u001b[m\u001b[K;\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:269:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfread(iGD->nTile, sizeof(int32_t)*m, 1, fp)\u001b[m\u001b[K;\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:278:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "      \u001b[01;35m\u001b[Kfread(iGD->nCnt[i], sizeof(int32_t)*k, 1, fp)\u001b[m\u001b[K;\n",
      "      \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:290:3:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "   \u001b[01;35m\u001b[Kfread(iGD->cName[i], 40, 1, fp)\u001b[m\u001b[K;\n",
      "   \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kigd_save\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:429:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "      \u001b[01;35m\u001b[Kfread(gdata, gdsize, 1, fp0)\u001b[m\u001b[K;\n",
      "      \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kigd0_save\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_base.c:487:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "      \u001b[01;35m\u001b[Kfread(gdata, gdsize, 1, fp0)\u001b[m\u001b[K;\n",
      "      \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "cc -c -g -O2 -lz -lm src/igd_create.c -o obj/igd_create.o \n",
      "cc -c -g -O2 -lz -lm src/igd_search.c -o obj/igd_search.o \n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kget_overlaps0\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:47:4:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "    \u001b[01;35m\u001b[Kfread(gData0, sizeof(gdata0_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "    \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:79:7:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "       \u001b[01;35m\u001b[Kfread(gData0, sizeof(gdata0_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "       \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kget_overlaps\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:157:4:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "    \u001b[01;35m\u001b[Kfread(gData, sizeof(gdata_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "    \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:189:7:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "       \u001b[01;35m\u001b[Kfread(gData, sizeof(gdata_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "       \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kget_overlaps_v\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:239:4:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "    \u001b[01;35m\u001b[Kfread(gData, sizeof(gdata_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "    \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:267:6:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "      \u001b[01;35m\u001b[Kfread(gData, sizeof(gdata_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "      \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KgetMap\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:363:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfread(gData, sizeof(gdata_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[KgetMap_v\u001b[m\u001b[K’:\n",
      "\u001b[01m\u001b[Ksrc/igd_search.c:420:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kfread\u001b[m\u001b[K’, declared with attribute warn_unused_result [\u001b[01;35m\u001b[K-Wunused-result\u001b[m\u001b[K]\n",
      "     \u001b[01;35m\u001b[Kfread(gData, sizeof(gdata_t)*tmpi, 1, fP)\u001b[m\u001b[K;\n",
      "     \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
      "cc -c -g -O2 -lz -lm src/igd.c -o obj/igd.o \n",
      "cc -o bin/igd obj/igd_base.o obj/igd_create.o obj/igd_search.o obj/igd.o -g -O2 -lz -lm\n"
     ]
    }
   ],
   "source": [
    "git clone git@github.com:databio/iGD\n",
    "cd iGD\n",
    "make\n",
    "\n",
    "#Add iGD bin to PATH (might have to do this before starting the tutorial) Something like \n",
    "export PATH=$BEDBASEtutorial/iGD/bin/:$PATH\n",
    "source ~/.bashrc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Run bedbuncher using Looper \n",
    "\n",
    "Once we have cloned the `bedbuncher` repository, set our local elasticsearch cluster and created the `iGD` binary, we can run `bedbuncher` passing the location of the `bedbase` configuration file to the argument `--bedbase-config`. Note: if the path to the `bedbase` configration file has been stored in the `$BEDBASE` environment variable, it's not neccesary to pass the `--bedbase-config` argument. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command: run (Looper version: 0.12.4)\n",
      "Reading sample table: '/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/bedbuncher_query.csv'\n",
      "Activating compute package 'local'\n",
      "Finding pipelines for protocol(s): bedbuncher\n",
      "Known protocols: bedbuncher\n",
      "'/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedbuncher/bedbuncher.py' appears to attempt to run on import; does it lack a conditional on '__main__'? Using base type: Sample\n",
      "\u001b[36m## [1 of 1] bedset1 (bedbuncher)\u001b[0m\n",
      "> Note (missing optional attribute): 'bedbuncher' requests sample attribute 'bbconfig_path' for option '--bedbase-config'\n",
      "Writing script to /home/jev4xy/Desktop/bedbase_tutorial/bedbuncher/bedbuncher_pipeline_logs/submission/bedbuncher_bedset1.sub\n",
      "Job script (n=1; 0.00 Gb): bedbuncher/bedbuncher_pipeline_logs/submission/bedbuncher_bedset1.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-19 18:39:50\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbase_demo_PEPs/../bedbuncher/bedbuncher.py --JSON-query-path bedbuncher/tests/test_query.json --bedset-name bedbase_demo_bedset -O bedbuncher/bedbuncher_pipeline_logs/results_pipeline -M 12000 --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml -R`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedbase_tutorial\n",
      "*            Outfolder:  bedbuncher/bedbuncher_pipeline_logs/results_pipeline/\n",
      "*  Pipeline started at:   (03-19 18:39:51) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedbase_tutorial/bedbuncher`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  3606a73841cb842f7408b67db8467f92e741fed5\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-13 13:40:45 -0400\n",
      "*        Pipeline diff:  1 file changed, 1 insertion(+), 1 deletion(-)\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*    `JSON_query_path`:  `bedbuncher/tests/test_query.json`\n",
      "*     `bedbase_config`:  `bedbase_demo_PEPs/bedbase_configuration.yaml`\n",
      "*        `bedset_name`:  `bedbase_demo_bedset`\n",
      "*        `config_file`:  `bedbuncher.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `12000`\n",
      "*          `new_start`:  `False`\n",
      "*      `output_parent`:  `bedbuncher/bedbuncher_pipeline_logs/results_pipeline`\n",
      "*            `recover`:  `True`\n",
      "*             `silent`:  `False`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Established connection with Elasticsearch: localhost\n",
      "Removed existing flag: 'bedbuncher/bedbuncher_pipeline_logs/results_pipeline/bedbuncher_completed.flag'\n",
      "14 BED files match the query\n",
      "bedset digest: 4b67b56dcbc2e13d161be7f8cf52d68b\n",
      "Output directory does not exist. Creating: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b\n",
      "Creating PEP annotation sheet and config.yaml for bedbase_demo_bedset\n",
      "Creating TAR archive: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset.tar\n",
      "bedbase_BEDfiles/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF937CGY_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE91663_ENCFF316ASR_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSM2423312_ENCFF155HVK_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSM2423313_ENCFF722AOG_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSM2827349_ENCFF196DNQ_peaks_GRCh38.bed.gz\n",
      "bedbase_BEDfiles/GSE105587_ENCFF413ANK_peaks_hg19.bed.gz\n",
      "bedbase_BEDfiles/GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19.bed.gz\n",
      "bedbase_BEDfiles/GSE105977_ENCFF634NTU_peaks_hg19.bed.gz\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_PEP\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_PEP/bedbase_demo_bedset_annotation_sheet.csv\n",
      "Reading individual BED file statistics from Elasticsearch\n",
      "Processing: GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38\n",
      "Processing: GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38\n",
      "Processing: GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38\n",
      "Processing: GSE105977_ENCFF937CGY_peaks_GRCh38\n",
      "Processing: GSE91663_ENCFF316ASR_peaks_GRCh38\n",
      "Processing: GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38\n",
      "Processing: GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38\n",
      "Processing: GSM2423312_ENCFF155HVK_peaks_GRCh38\n",
      "Processing: GSM2423313_ENCFF722AOG_peaks_GRCh38\n",
      "Processing: GSM2827349_ENCFF196DNQ_peaks_GRCh38\n",
      "Processing: GSE105587_ENCFF413ANK_peaks_hg19\n",
      "Processing: GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19\n",
      "Processing: GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19\n",
      "Processing: GSE105977_ENCFF634NTU_peaks_hg19\n",
      "Calculating bedset statistics\n",
      "Saving bedfiles statistics to: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_bedstat.csv\n",
      "Saving bedset statistics to: /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_summaryStats.csv\n",
      "Creating iGD database\n",
      "Target to produce: `/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_igd.tar.gz`  \n",
      "\n",
      "> `igd create /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset.txt /home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_igd bedbase_demo_bedset -f` (17993)\n",
      "<pre>\n",
      "igd_create 0\n",
      "igd_create 1: 0\n",
      "igd_create 2\n",
      "igd_create 3\n",
      "igd_create 4\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:00. Running peak memory: 0GB.  \n",
      "  PID: 17993;\tCommand: igd;\tReturn code: 0;\tMemory used: 0.0GB\n",
      "\n",
      "Creating iGD database TAR archive: bedbase_demo_bedset_igd.tar.gz\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_igd\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_igd/bedbase_demo_bedset_index.tsv\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_igd/data0\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_igd/bedbase_demo_bedset.igd\n",
      "Creating PEP TAR archive: bedbase_demo_bedset_PEP.tar.gz\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_PEP\n",
      "/home/jev4xy/Desktop/bedbase_tutorial/bedstat/bedstat_output/4b67b56dcbc2e13d161be7f8cf52d68b/bedbase_demo_bedset_PEP/bedbase_demo_bedset_annotation_sheet.csv\n",
      "{'GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38': '78c0e4753d04b238fc07e4ebe5a02984', 'GSE105977_ENCFF617QGK_optimal_idr_thresholded_peaks_GRCh38': 'fdd94ac0787599d564b07193e4ec41fd', 'GSE105977_ENCFF793SZW_conservative_idr_thresholded_peaks_GRCh38': 'a6a08126cb6f4b1953ba0ec8675df85a', 'GSE105977_ENCFF937CGY_peaks_GRCh38': 'a78493a2b314afe9f6635c4883f0d44b', 'GSE91663_ENCFF316ASR_peaks_GRCh38': '50e19bd44174bb286aa28ae2a15e7b8f', 'GSE91663_ENCFF319TPR_conservative_idr_thresholded_peaks_GRCh38': '9cd65cf4f07b83af35770c4a098fd4c6', 'GSE91663_ENCFF553KIK_optimal_idr_thresholded_peaks_GRCh38': 'a5af5857bfbc3bfc8fea09cb90e67a16', 'GSM2423312_ENCFF155HVK_peaks_GRCh38': '02fd518818560c28ed20ed98f4c291bd', 'GSM2423313_ENCFF722AOG_peaks_GRCh38': '33d4328fe4ff3a472edff81bf8f5d566', 'GSM2827349_ENCFF196DNQ_peaks_GRCh38': '2ffb2cedd14f5f1fae7cb765a66d82a3', 'GSE105587_ENCFF413ANK_peaks_hg19': 'f41e12ddd3b6c4ee6da2140d0feee1ea', 'GSE105587_ENCFF809OOE_conservative_idr_thresholded_peaks_hg19': '9dc6f420639e0a265f3f179b6b42713a', 'GSE105977_ENCFF449EZT_optimal_idr_thresholded_peaks_hg19': 'e577eb947b5c791b30df969f0564324b', 'GSE105977_ENCFF634NTU_peaks_hg19': '021461e2edc7b85315042c5ef80c7c03'}\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'bedbase_demo_bedset' summary info was successfully inserted into bedsets\n",
      "Starting cleanup: 3 files; 0 conditional files for cleanup\n",
      "\n",
      "Cleaning up flagged intermediate files. . .\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:00\n",
      "*  Total elapsed time (all runs):  0:00:00\n",
      "*         Peak memory (this run):  0 GB\n",
      "*        Pipeline completed time: 2020-03-19 18:39:51\n",
      "\n",
      "Looper finished\n",
      "Samples valid for job generation: 1 of 1\n",
      "Successful samples: 1 of 1\n",
      "Commands submitted: 1 of 1\n",
      "Jobs submitted: 1\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "looper run  bedbase_demo_PEPs/bedbuncher_config.yaml  --bedbase-config bedbase_demo_PEPs/bedbase_configuration.yaml \\\n",
    "--compute local -R"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Run local instance of the bedhost API\n",
    "\n",
    "The last part of the tutorial consists on running a local instance of [bedhost](https://github.com/databio/bedhost/tree/master) (a REST API for bedstat and bedbuncher produced outputs) in order to explore plots, statistics and download pipeline outputs. To run `bedhost`, we can clone the github repository and pip install the package as follows:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cloning into 'bedhost'...\n",
      "remote: Enumerating objects: 111, done.\u001b[K\n",
      "remote: Counting objects: 100% (111/111), done.\u001b[K\n",
      "remote: Compressing objects: 100% (81/81), done.\u001b[K\n",
      "remote: Total 622 (delta 72), reused 66 (delta 29), pack-reused 511\u001b[K\n",
      "Receiving objects: 100% (622/622), 178.95 KiB | 1.41 MiB/s, done.\n",
      "Resolving deltas: 100% (405/405), done.\n",
      "Processing ./bedhost\n",
      "Requirement already satisfied: aiofiles in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (0.4.0)\n",
      "Requirement already satisfied: bbconf>=0.0.2-dev in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (0.0.2.dev0)\n",
      "Requirement already satisfied: elasticsearch-dsl<8.0.0,>=7.0.0 in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (7.1.0)\n",
      "Requirement already satisfied: fastapi in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (0.49.0)\n",
      "Requirement already satisfied: jinja2 in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (2.10.1)\n",
      "Requirement already satisfied: python-multipart in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (0.0.5)\n",
      "Requirement already satisfied: pyyaml in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (5.1.2)\n",
      "Requirement already satisfied: requests in /usr/lib/python3/dist-packages (from bedhost==0.0.1) (2.18.4)\n",
      "Requirement already satisfied: starlette in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (0.12.9)\n",
      "Requirement already satisfied: uvicorn>=0.7.1 in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (0.11.3)\n",
      "Requirement already satisfied: yacman in /home/jev4xy/.local/lib/python3.6/site-packages (from bedhost==0.0.1) (0.6.6)\n",
      "Requirement already satisfied: elasticsearch in /home/jev4xy/.local/lib/python3.6/site-packages (from bbconf>=0.0.2-dev->bedhost==0.0.1) (7.1.0)\n",
      "Requirement already satisfied: logmuse in /home/jev4xy/.local/lib/python3.6/site-packages (from bbconf>=0.0.2-dev->bedhost==0.0.1) (0.2.5)\n",
      "Requirement already satisfied: six in /home/jev4xy/.local/lib/python3.6/site-packages (from elasticsearch-dsl<8.0.0,>=7.0.0->bedhost==0.0.1) (1.12.0)\n",
      "Requirement already satisfied: python-dateutil in /home/jev4xy/.local/lib/python3.6/site-packages (from elasticsearch-dsl<8.0.0,>=7.0.0->bedhost==0.0.1) (2.8.0)\n",
      "Requirement already satisfied: pydantic<2.0.0,>=0.32.2 in /home/jev4xy/.local/lib/python3.6/site-packages (from fastapi->bedhost==0.0.1) (1.4)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /home/jev4xy/.local/lib/python3.6/site-packages (from jinja2->bedhost==0.0.1) (1.1.1)\n",
      "Requirement already satisfied: h11<0.10,>=0.8 in /home/jev4xy/.local/lib/python3.6/site-packages (from uvicorn>=0.7.1->bedhost==0.0.1) (0.9.0)\n",
      "Requirement already satisfied: websockets==8.* in /home/jev4xy/.local/lib/python3.6/site-packages (from uvicorn>=0.7.1->bedhost==0.0.1) (8.1)\n",
      "Requirement already satisfied: uvloop>=0.14.0; sys_platform != \"win32\" and sys_platform != \"cygwin\" and platform_python_implementation != \"PyPy\" in /home/jev4xy/.local/lib/python3.6/site-packages (from uvicorn>=0.7.1->bedhost==0.0.1) (0.14.0)\n",
      "Requirement already satisfied: click==7.* in /home/jev4xy/.local/lib/python3.6/site-packages (from uvicorn>=0.7.1->bedhost==0.0.1) (7.0)\n",
      "Requirement already satisfied: httptools==0.1.*; sys_platform != \"win32\" and sys_platform != \"cygwin\" and platform_python_implementation != \"PyPy\" in /home/jev4xy/.local/lib/python3.6/site-packages (from uvicorn>=0.7.1->bedhost==0.0.1) (0.1.1)\n",
      "Requirement already satisfied: oyaml in /home/jev4xy/.local/lib/python3.6/site-packages (from yacman->bedhost==0.0.1) (0.9)\n",
      "Requirement already satisfied: ubiquerg>=0.4.9 in /home/jev4xy/.local/lib/python3.6/site-packages (from yacman->bedhost==0.0.1) (0.5.0)\n",
      "Requirement already satisfied: attmap>=0.12.9 in /home/jev4xy/.local/lib/python3.6/site-packages (from yacman->bedhost==0.0.1) (0.12.9)\n",
      "Requirement already satisfied: urllib3>=1.21.1 in /usr/lib/python3/dist-packages (from elasticsearch->bbconf>=0.0.2-dev->bedhost==0.0.1) (1.22)\n",
      "Requirement already satisfied: dataclasses>=0.6; python_version < \"3.7\" in /home/jev4xy/.local/lib/python3.6/site-packages (from pydantic<2.0.0,>=0.32.2->fastapi->bedhost==0.0.1) (0.7)\n",
      "Building wheels for collected packages: bedhost\n",
      "  Building wheel for bedhost (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for bedhost: filename=bedhost-0.0.1-cp36-none-any.whl size=59799 sha256=3da5288dcd0f404fc58e0e4a6ebc382e7410619abd6657ff37a5fc0f8961709f\n",
      "  Stored in directory: /tmp/pip-ephem-wheel-cache-h0h97mve/wheels/0d/13/b6/f9f990b04e991dfbb802fbdb6628b11149fedfb88a6916dfe0\n",
      "Successfully built bedhost\n",
      "Installing collected packages: bedhost\n",
      "  Found existing installation: bedhost 0.0.1\n",
      "    Uninstalling bedhost-0.0.1:\n",
      "      Successfully uninstalled bedhost-0.0.1\n",
      "Successfully installed bedhost-0.0.1\n",
      "\u001b[33mWARNING: You are using pip version 19.3.1; however, version 20.0.2 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "git clone git@github.com:databio/bedhost\n",
    "pip install bedhost/. --user"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To start bedhost, we simply need to run the following commands passing the location of the `bedbase` config file to the `-c` flag.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: itsdangerous in /home/jev4xy/.local/lib/python3.6/site-packages (1.1.0)\n",
      "\u001b[33mWARNING: You are using pip version 19.3.1; however, version 20.0.2 is available.\n",
      "You should consider upgrading via the 'pip install --upgrade pip' command.\u001b[0m\n",
      "DEBU 2020-03-19 18:45:30,471 | bedhost:est:263 > Configured logger 'bedhost' using logmuse v0.2.5 \n",
      "DEBU 18:45:30 | bbconf:est:263 > Configured logger 'bbconf' using logmuse v0.2.5 \n",
      "INFO 18:45:30 | bbconf:bbconf:58 > Established connection with Elasticsearch: localhost \n",
      "DEBU 18:45:30 | bbconf:bbconf:59 > Elasticsearch info:\n",
      "{'name': '1ec537ca3e87', 'cluster_name': 'docker-cluster', 'cluster_uuid': 'PamppPmESrKNFL1hqlo6gA', 'version': {'number': '7.5.1', 'build_flavor': 'default', 'build_type': 'docker', 'build_hash': '3ae9ac9a93c95bd0cdc054951cf95d88e1e18d96', 'build_date': '2019-12-16T22:57:37.835892Z', 'build_snapshot': False, 'lucene_version': '8.3.0', 'minimum_wire_compatibility_version': '6.8.0', 'minimum_index_compatibility_version': '6.0.0-beta1'}, 'tagline': 'You Know, for Search'} \n",
      "Traceback (most recent call last):\n",
      "  File \"/home/jev4xy/.local/bin/bedhost\", line 8, in <module>\n",
      "    sys.exit(main())\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/bedhost/main.py\", line 253, in main\n",
      "    StaticFiles(directory=bbc[CFG_PATH_KEY][CFG_PIP_OUTPUT_KEY]), name=BED_INDEX)\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/starlette/applications.py\", line 41, in mount\n",
      "    self.router.mount(path, app=app, name=name)\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/starlette/routing.py\", line 499, in mount\n",
      "    route = Mount(path, app=app, name=name)\n",
      "  File \"/home/jev4xy/.local/lib/python3.6/site-packages/starlette/routing.py\", line 282, in __init__\n",
      "    assert path == \"\" or path.startswith(\"/\"), \"Routed paths must start with '/'\"\n",
      "AssertionError: Routed paths must start with '/'\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#Install bedhost dependencies\n",
    "pip install itsdangerous --user\n",
    "\n",
    "bedhost serve -c  bedbase_demo_PEPs/bedbase_configuration.yaml\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If we have stored the path to the bedbase config in the environment variable `$BEDBASE` (suggested), it's not neccesary to pass the `-c` flag. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bedhost serve "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": true,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": true,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "329.797px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
