{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# **BEDHOST Demo**\n",
    "\n",
    "The following demo has the purpose of demonstrating how to process, generate statistics and plots of BED files genrated by the R package Genomic Distributions using the REST API for the bedstat and bedbuncher pipelines. \n",
    "\n",
    "The general workflow for uploading bed files and their \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## To get started:\n",
    "\n",
    "In order to get started, we'll need a PEP [Portable Encapsulated project](https://pepkit.github.io/) using derived attributes to point to the BED files as well as their metadata. The following is an example of a config file using the derived attributes output_file_path and yaml_file to point to the `.bed.gz` files and their respective metadata."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metadata:\n",
      "  sample_table: demo_annotation_sheet.csv\n",
      "  output_dir: $HOME/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results \n",
      "  pipeline_interfaces: ../../pipeline_interface.yaml\n",
      "\n",
      "constant_attributes: \n",
      "  output_file_path: \"source\"\n",
      "  yaml_file: \"source2\"\n",
      "  protocol: \"bedstat\"\n",
      "\n",
      "derived_attributes: [output_file_path, yaml_file]\n",
      "data_sources:\n",
      "  source: \"../{file_name}\" \n",
      "  source2: \"$HOME/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results/submission/{sample_name}.yaml\""
     ]
    }
   ],
   "source": [
    "cat demo_config.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "To run the [bedstat](https://github.com/databio/bedstat) pipeline, clone the github repository. For detailed instructions in how to link a project to a pipeline, click [here](http://looper.databio.org/en/latest/linking-a-pipeline/)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Command: run (Looper version: 0.12.4)\n",
      "Reading sample table: '/home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/bedhost_demo_refPEP/demo_annotation_sheet.csv'\n",
      "Activating compute package 'local'\n",
      "Finding pipelines for protocol(s): bedstat\n",
      "Known protocols: bedstat\n",
      "'pipeline/bedstat.py' appears to attempt to run on import; does it lack a conditional on '__main__'? Using base type: Sample\n",
      "\u001b[36m## [1 of 15] bedhost_demo_db1 (bedstat)\u001b[0m\n",
      "Submission settings lack memory specification\n",
      "Writing script to /home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results/submission/bedstat_bedhost_demo_db1.sub\n",
      "Job script (n=1; 0.00 Gb): /home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results/submission/bedstat_bedhost_demo_db1.sub\n",
      "Compute node: cphg-51ksmr2\n",
      "Start time: 2020-03-13 17:53:03\n",
      "### Pipeline run code and environment:\n",
      "\n",
      "*              Command:  `pipeline/bedstat.py --bedfile /home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz --genome hg38 --sample-yaml /home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results/submission/bedhost_demo_db1.yaml -O /home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results/results_pipeline --no-db-commit`\n",
      "*         Compute host:  cphg-51ksmr2\n",
      "*          Working dir:  /home/jev4xy/Desktop/bedstat\n",
      "*            Outfolder:  /home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/\n",
      "*  Pipeline started at:   (03-13 17:53:03) elapsed: 0.0 _TIME_\n",
      "\n",
      "### Version log:\n",
      "\n",
      "*       Python version:  3.6.8\n",
      "*          Pypiper dir:  `/home/jev4xy/.local/lib/python3.6/site-packages/pypiper`\n",
      "*      Pypiper version:  0.12.1\n",
      "*         Pipeline dir:  `/home/jev4xy/Desktop/bedstat/pipeline`\n",
      "*     Pipeline version:  None\n",
      "*        Pipeline hash:  94ef6148e2260b392c97e25b9853fbc197770e16\n",
      "*      Pipeline branch:  * master\n",
      "*        Pipeline date:  2020-03-13 11:57:56 -0400\n",
      "*        Pipeline diff:  1 file changed, 1 deletion(-)\n",
      "\n",
      "### Arguments passed to pipeline:\n",
      "\n",
      "*     `bedbase_config`:  `None`\n",
      "*            `bedfile`:  `/home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz`\n",
      "*        `config_file`:  `bedstat.yaml`\n",
      "*              `cores`:  `1`\n",
      "*              `dirty`:  `False`\n",
      "*       `force_follow`:  `False`\n",
      "*    `genome_assembly`:  `hg38`\n",
      "*              `input`:  `None`\n",
      "*             `input2`:  `None`\n",
      "*     `just_db_commit`:  `False`\n",
      "*             `logdev`:  `False`\n",
      "*                `mem`:  `4000`\n",
      "*          `new_start`:  `False`\n",
      "*       `no_db_commit`:  `True`\n",
      "*      `output_parent`:  `/home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results/results_pipeline`\n",
      "*            `recover`:  `False`\n",
      "*        `sample_name`:  `None`\n",
      "*        `sample_yaml`:  `/home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/bedstat_pipeline_results/submission/bedhost_demo_db1.yaml`\n",
      "*             `silent`:  `False`\n",
      "*   `single_or_paired`:  `single`\n",
      "*           `testmode`:  `False`\n",
      "*          `verbosity`:  `None`\n",
      "\n",
      "----------------------------------------\n",
      "\n",
      "Target to produce: `/home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.json`  \n",
      "\n",
      "> `Rscript /home/jev4xy/Desktop/bedstat/tools/regionstat.R --bedfile=/home/jev4xy/Desktop/bedstat/bedhost_demo_files_justBED/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38.bed.gz --fileId=GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38 --outputfolder=/home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984 --genome=hg38 --digest=78c0e4753d04b238fc07e4ebe5a02984` (24402)\n",
      "<pre>\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_tssdist\"\n",
      "BSAggregate: Calculating sizes. You can speed this up by supplying a regionsGRL.length vector...Done counting regionsGRL lengths.\n",
      "Finding overlaps...\n",
      "Setting regionIDs...\n",
      "jExpr: .N\n",
      "Combining...\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_chrombins\"\n",
      "Loading required namespace: BSgenome.Hsapiens.UCSC.hg38.masked\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_gccontent\"\n",
      "promoterCore :\tfound 11017\n",
      "promoterProx :\tfound 2911\n",
      "exon :\tfound 11031\n",
      "intron :\tfound 22553\n",
      "[1] \"Plotting: /home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/GSE105587_ENCFF018NNF_conservative_idr_thresholded_peaks_GRCh38_partitions\"\n",
      "Warning messages:\n",
      "1: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n",
      "2: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n",
      "4: In .Seqinfo.mergexy(x, y) :\n",
      "  Each of the 2 combined objects has sequence levels not in the other:\n",
      "  - in 'x': chrUn_GL000224v1, chr17_GL000205v2_random, chrUn_GL000219v1, chrUn_GL000195v1, chrUn_GL000218v1, chr22_KI270733v1_random, chr1_KI270706v1_random, chrUn_GL000220v1, chrUn_GL000216v2, chr17_KI270729v1_random, chr1_KI270713v1_random\n",
      "  - in 'y': chrCHR_HG107_PATCH, chrCHR_HG126_PATCH, chrCHR_HG1311_PATCH, chrCHR_HG1342_HG2282_PATCH, chrCHR_HG1362_PATCH, chrCHR_HG142_HG150_NOVEL_TEST, chrCHR_HG151_NOVEL_TEST, chrCHR_HG1832_PATCH, chrCHR_HG2021_PATCH, chrCHR_HG2023_PATCH, chrCHR_HG2030_PATCH, chrCHR_HG2058_PATCH, chrCHR_HG2063_PATCH, chrCHR_HG2066_PATCH, chrCHR_HG2072_PATCH, chrCHR_HG2095_PATCH, chrCHR_HG2104_PATCH, chrCHR_HG2116_PATCH, chrCHR_HG2191_PATCH, chrCHR_HG2213_PATCH, chrCHR_HG2217_PATCH, chrCHR_HG2232_PATCH, chrCHR_HG2233_PATCH, chrCHR_HG2235_PATCH, chrCHR_HG2239_PATCH, chrCHR_HG2247_PATCH, chrCHR_HG2288_HG2289_PATCH, chrCHR_HG2290_PATCH, chrCHR_HG2291_PATCH, chrCHR_HG2334_PATCH, chrCHR_HG26_PATCH, ch [... truncated]\n",
      "</pre>\n",
      "Command completed. Elapsed time: 0:00:23. Running peak memory: 0.539GB.  \n",
      "  PID: 24402;\tCommand: Rscript;\tReturn code: 0;\tMemory used: 0.539GB\n",
      "\n",
      "\n",
      "### Pipeline completed. Epilogue\n",
      "*        Elapsed time (this run):  0:00:23\n",
      "*  Total elapsed time (all runs):  0:00:23\n",
      "*         Peak memory (this run):  0.5389 GB\n",
      "*        Pipeline completed time: 2020-03-13 17:53:26\n",
      "Removed existing flag: '/home/jev4xy/Desktop/rivanna_project_sshfs/resources/regions/bedstat_output/78c0e4753d04b238fc07e4ebe5a02984/bedstat-pipeline_completed.flag'\n",
      "\n",
      "Looper finished\n",
      "Samples valid for job generation: 1 of 1\n",
      "Successful samples: 1 of 1\n",
      "Commands submitted: 1 of 1\n",
      "Jobs submitted: 1\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "cd ~/Desktop/bedstat\n",
    "looper run bedhost_demo_files_justBED/bedhost_demo_refPEP/demo_config.yaml --no-db-commit \\\n",
    "--compute local --limit 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
